// ts-jest/dist/transpiler.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var fs = require("fs");
var process_1 = require("process");
var ts = require("typescript");
var logger_1 = require("./logger");
function transpileTypescript(filePath, fileSrc, compilerOptions, tsJestConfig) {
    if (tsJestConfig.useExperimentalLanguageServer) {
        logger_1.logOnce('Using experimental language server.');
        return transpileViaLanguageServer(filePath, fileSrc, compilerOptions);
    }
    logger_1.logOnce('Compiling via normal transpileModule call');
    return transpileViaTranspileModile(filePath, fileSrc, compilerOptions);
}
exports.transpileTypescript = transpileTypescript;
function transpileViaLanguageServer(filePath, fileSrc, compilerOptions) {
    var serviceHost = {
        getScriptFileNames: function () {
            return [filePath];
        },
        getScriptVersion: function (fileName) {
            return undefined;
        },
        getCurrentDirectory: function () {
            return process_1.cwd();
        },
        getScriptSnapshot: function (fileName) {
            if (fileName === filePath) {
                return ts.ScriptSnapshot.fromString(fileSrc);
            }
            var result = fs.readFileSync(fileName, 'utf8');
            return ts.ScriptSnapshot.fromString(result);
        },
        getCompilationSettings: function () {
            return compilerOptions;
        },
        getDefaultLibFileName: function () {
            return ts.getDefaultLibFilePath(compilerOptions);
        },
        fileExists: ts.sys.fileExists,
        readFile: ts.sys.readFile,
        readDirectory: ts.sys.readDirectory,
        realpath: ts.sys.realpath,
        getDirectories: ts.sys.getDirectories,
        directoryExists: ts.sys.directoryExists,
    };
    var service = ts.createLanguageService(serviceHost);
    var serviceOutput = service.getEmitOutput(filePath);
    var files = serviceOutput.outputFiles.filter(function (file) {
        return file.name.endsWith('js');
    });
    logger_1.logOnce('JS files parsed', files.map(function (f) { return f.name; }));
    var diagnostics = service
        .getCompilerOptionsDiagnostics()
        .concat(service.getSyntacticDiagnostics(filePath))
        .concat(service.getSemanticDiagnostics(filePath));
    if (diagnostics.length > 0) {
        var errors = diagnostics.map(function (d) { return d.messageText; }) + "\n";
        logger_1.logOnce("Diagnostic errors from TSC: " + errors);
        throw Error("TSC language server encountered errors while transpiling. Errors: " + errors);
    }
    return files[0].text;
}
function transpileViaTranspileModile(filePath, fileSource, compilerOptions) {
    return ts.transpileModule(fileSource, {
        compilerOptions: compilerOptions,
        fileName: filePath,
    }).outputText;
}
