// @aws-sdk/client-cognito-identity-provider/dist-cjs/schemas/schemas_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAuthnChallengeNotFoundException$ = exports.UserPoolTaggingException$ = exports.UserPoolAddOnNotEnabledException$ = exports.UserNotFoundException$ = exports.UserNotConfirmedException$ = exports.UsernameExistsException$ = exports.UserLambdaValidationException$ = exports.UserImportInProgressException$ = exports.UnsupportedUserStateException$ = exports.UnsupportedTokenTypeException$ = exports.UnsupportedOperationException$ = exports.UnsupportedIdentityProviderException$ = exports.UnexpectedLambdaException$ = exports.UnauthorizedException$ = exports.TooManyRequestsException$ = exports.TooManyFailedAttemptsException$ = exports.TierChangeNotAllowedException$ = exports.TermsExistsException$ = exports.SoftwareTokenMFANotFoundException$ = exports.ScopeDoesNotExistException$ = exports.ResourceNotFoundException$ = exports.RefreshTokenReuseException$ = exports.PreconditionNotMetException$ = exports.PasswordResetRequiredException$ = exports.PasswordHistoryPolicyViolationException$ = exports.NotAuthorizedException$ = exports.MFAMethodNotFoundException$ = exports.ManagedLoginBrandingExistsException$ = exports.LimitExceededException$ = exports.InvalidUserPoolConfigurationException$ = exports.InvalidSmsRoleTrustRelationshipException$ = exports.InvalidSmsRoleAccessPolicyException$ = exports.InvalidPasswordException$ = exports.InvalidParameterException$ = exports.InvalidOAuthFlowException$ = exports.InvalidLambdaResponseException$ = exports.InvalidEmailRoleAccessPolicyException$ = exports.InternalErrorException$ = exports.GroupExistsException$ = exports.ForbiddenException$ = exports.FeatureUnavailableInTierException$ = exports.ExpiredCodeException$ = exports.EnableSoftwareTokenMFAException$ = exports.DuplicateProviderException$ = exports.DeviceKeyExistsException$ = exports.ConcurrentModificationException$ = exports.CodeMismatchException$ = exports.CodeDeliveryFailureException$ = exports.AliasExistsException$ = exports.CognitoIdentityProviderServiceException$ = void 0;
exports.AdminSetUserMFAPreferenceResponse$ = exports.AdminSetUserMFAPreferenceRequest$ = exports.AdminRespondToAuthChallengeResponse$ = exports.AdminRespondToAuthChallengeRequest$ = exports.AdminResetUserPasswordResponse$ = exports.AdminResetUserPasswordRequest$ = exports.AdminRemoveUserFromGroupRequest$ = exports.AdminListUserAuthEventsResponse$ = exports.AdminListUserAuthEventsRequest$ = exports.AdminListGroupsForUserResponse$ = exports.AdminListGroupsForUserRequest$ = exports.AdminListDevicesResponse$ = exports.AdminListDevicesRequest$ = exports.AdminLinkProviderForUserResponse$ = exports.AdminLinkProviderForUserRequest$ = exports.AdminInitiateAuthResponse$ = exports.AdminInitiateAuthRequest$ = exports.AdminGetUserResponse$ = exports.AdminGetUserRequest$ = exports.AdminGetDeviceResponse$ = exports.AdminGetDeviceRequest$ = exports.AdminForgetDeviceRequest$ = exports.AdminEnableUserResponse$ = exports.AdminEnableUserRequest$ = exports.AdminDisableUserResponse$ = exports.AdminDisableUserRequest$ = exports.AdminDisableProviderForUserResponse$ = exports.AdminDisableProviderForUserRequest$ = exports.AdminDeleteUserRequest$ = exports.AdminDeleteUserAttributesResponse$ = exports.AdminDeleteUserAttributesRequest$ = exports.AdminCreateUserResponse$ = exports.AdminCreateUserRequest$ = exports.AdminCreateUserConfigType$ = exports.AdminConfirmSignUpResponse$ = exports.AdminConfirmSignUpRequest$ = exports.AdminAddUserToGroupRequest$ = exports.AddCustomAttributesResponse$ = exports.AddCustomAttributesRequest$ = exports.AccountTakeoverRiskConfigurationType$ = exports.AccountTakeoverActionType$ = exports.AccountTakeoverActionsType$ = exports.AccountRecoverySettingType$ = exports.errorTypeRegistries = exports.WebAuthnRelyingPartyMismatchException$ = exports.WebAuthnOriginNotAllowedException$ = exports.WebAuthnNotEnabledException$ = exports.WebAuthnCredentialNotSupportedException$ = exports.WebAuthnConfigurationMissingException$ = exports.WebAuthnClientMismatchException$ = void 0;
exports.CreateUserPoolClientRequest$ = exports.CreateUserImportJobResponse$ = exports.CreateUserImportJobRequest$ = exports.CreateTermsResponse$ = exports.CreateTermsRequest$ = exports.CreateResourceServerResponse$ = exports.CreateResourceServerRequest$ = exports.CreateManagedLoginBrandingResponse$ = exports.CreateManagedLoginBrandingRequest$ = exports.CreateIdentityProviderResponse$ = exports.CreateIdentityProviderRequest$ = exports.CreateGroupResponse$ = exports.CreateGroupRequest$ = exports.ContextDataType$ = exports.ConfirmSignUpResponse$ = exports.ConfirmSignUpRequest$ = exports.ConfirmForgotPasswordResponse$ = exports.ConfirmForgotPasswordRequest$ = exports.ConfirmDeviceResponse$ = exports.ConfirmDeviceRequest$ = exports.CompromisedCredentialsRiskConfigurationType$ = exports.CompromisedCredentialsActionsType$ = exports.CompleteWebAuthnRegistrationResponse$ = exports.CompleteWebAuthnRegistrationRequest$ = exports.CodeDeliveryDetailsType$ = exports.CloudWatchLogsConfigurationType$ = exports.ChangePasswordResponse$ = exports.ChangePasswordRequest$ = exports.ChallengeResponseType$ = exports.AuthEventType$ = exports.AuthenticationResultType$ = exports.AttributeType$ = exports.AssociateSoftwareTokenResponse$ = exports.AssociateSoftwareTokenRequest$ = exports.AssetType$ = exports.AnalyticsMetadataType$ = exports.AnalyticsConfigurationType$ = exports.AdvancedSecurityAdditionalFlowsType$ = exports.AdminUserGlobalSignOutResponse$ = exports.AdminUserGlobalSignOutRequest$ = exports.AdminUpdateUserAttributesResponse$ = exports.AdminUpdateUserAttributesRequest$ = exports.AdminUpdateDeviceStatusResponse$ = exports.AdminUpdateDeviceStatusRequest$ = exports.AdminUpdateAuthEventFeedbackResponse$ = exports.AdminUpdateAuthEventFeedbackRequest$ = exports.AdminSetUserSettingsResponse$ = exports.AdminSetUserSettingsRequest$ = exports.AdminSetUserPasswordResponse$ = exports.AdminSetUserPasswordRequest$ = void 0;
exports.EventContextDataType$ = exports.EmailMfaSettingsType$ = exports.EmailMfaConfigType$ = exports.EmailConfigurationType$ = exports.DomainDescriptionType$ = exports.DeviceType$ = exports.DeviceSecretVerifierConfigType$ = exports.DeviceConfigurationType$ = exports.DescribeUserPoolResponse$ = exports.DescribeUserPoolRequest$ = exports.DescribeUserPoolDomainResponse$ = exports.DescribeUserPoolDomainRequest$ = exports.DescribeUserPoolClientResponse$ = exports.DescribeUserPoolClientRequest$ = exports.DescribeUserImportJobResponse$ = exports.DescribeUserImportJobRequest$ = exports.DescribeTermsResponse$ = exports.DescribeTermsRequest$ = exports.DescribeRiskConfigurationResponse$ = exports.DescribeRiskConfigurationRequest$ = exports.DescribeResourceServerResponse$ = exports.DescribeResourceServerRequest$ = exports.DescribeManagedLoginBrandingResponse$ = exports.DescribeManagedLoginBrandingRequest$ = exports.DescribeManagedLoginBrandingByClientResponse$ = exports.DescribeManagedLoginBrandingByClientRequest$ = exports.DescribeIdentityProviderResponse$ = exports.DescribeIdentityProviderRequest$ = exports.DeleteWebAuthnCredentialResponse$ = exports.DeleteWebAuthnCredentialRequest$ = exports.DeleteUserRequest$ = exports.DeleteUserPoolRequest$ = exports.DeleteUserPoolDomainResponse$ = exports.DeleteUserPoolDomainRequest$ = exports.DeleteUserPoolClientRequest$ = exports.DeleteUserAttributesResponse$ = exports.DeleteUserAttributesRequest$ = exports.DeleteTermsRequest$ = exports.DeleteResourceServerRequest$ = exports.DeleteManagedLoginBrandingRequest$ = exports.DeleteIdentityProviderRequest$ = exports.DeleteGroupRequest$ = exports.CustomSMSLambdaVersionConfigType$ = exports.CustomEmailLambdaVersionConfigType$ = exports.CustomDomainConfigType$ = exports.CreateUserPoolResponse$ = exports.CreateUserPoolRequest$ = exports.CreateUserPoolDomainResponse$ = exports.CreateUserPoolDomainRequest$ = exports.CreateUserPoolClientResponse$ = void 0;
exports.ListTermsRequest$ = exports.ListTagsForResourceResponse$ = exports.ListTagsForResourceRequest$ = exports.ListResourceServersResponse$ = exports.ListResourceServersRequest$ = exports.ListIdentityProvidersResponse$ = exports.ListIdentityProvidersRequest$ = exports.ListGroupsResponse$ = exports.ListGroupsRequest$ = exports.ListDevicesResponse$ = exports.ListDevicesRequest$ = exports.LambdaConfigType$ = exports.InitiateAuthResponse$ = exports.InitiateAuthRequest$ = exports.InboundFederationLambdaType$ = exports.IdentityProviderType$ = exports.HttpHeader$ = exports.GroupType$ = exports.GlobalSignOutResponse$ = exports.GlobalSignOutRequest$ = exports.GetUserResponse$ = exports.GetUserRequest$ = exports.GetUserPoolMfaConfigResponse$ = exports.GetUserPoolMfaConfigRequest$ = exports.GetUserAuthFactorsResponse$ = exports.GetUserAuthFactorsRequest$ = exports.GetUserAttributeVerificationCodeResponse$ = exports.GetUserAttributeVerificationCodeRequest$ = exports.GetUICustomizationResponse$ = exports.GetUICustomizationRequest$ = exports.GetTokensFromRefreshTokenResponse$ = exports.GetTokensFromRefreshTokenRequest$ = exports.GetSigningCertificateResponse$ = exports.GetSigningCertificateRequest$ = exports.GetLogDeliveryConfigurationResponse$ = exports.GetLogDeliveryConfigurationRequest$ = exports.GetIdentityProviderByIdentifierResponse$ = exports.GetIdentityProviderByIdentifierRequest$ = exports.GetGroupResponse$ = exports.GetGroupRequest$ = exports.GetDeviceResponse$ = exports.GetDeviceRequest$ = exports.GetCSVHeaderResponse$ = exports.GetCSVHeaderRequest$ = exports.ForgotPasswordResponse$ = exports.ForgotPasswordRequest$ = exports.ForgetDeviceRequest$ = exports.FirehoseConfigurationType$ = exports.EventRiskType$ = exports.EventFeedbackType$ = void 0;
exports.SetUserPoolMfaConfigResponse$ = exports.SetUserPoolMfaConfigRequest$ = exports.SetUserMFAPreferenceResponse$ = exports.SetUserMFAPreferenceRequest$ = exports.SetUICustomizationResponse$ = exports.SetUICustomizationRequest$ = exports.SetRiskConfigurationResponse$ = exports.SetRiskConfigurationRequest$ = exports.SetLogDeliveryConfigurationResponse$ = exports.SetLogDeliveryConfigurationRequest$ = exports.SchemaAttributeType$ = exports.S3ConfigurationType$ = exports.RiskExceptionConfigurationType$ = exports.RiskConfigurationType$ = exports.RevokeTokenResponse$ = exports.RevokeTokenRequest$ = exports.RespondToAuthChallengeResponse$ = exports.RespondToAuthChallengeRequest$ = exports.ResourceServerType$ = exports.ResourceServerScopeType$ = exports.ResendConfirmationCodeResponse$ = exports.ResendConfirmationCodeRequest$ = exports.RefreshTokenRotationType$ = exports.RecoveryOptionType$ = exports.ProviderUserIdentifierType$ = exports.ProviderDescription$ = exports.PreTokenGenerationVersionConfigType$ = exports.PasswordPolicyType$ = exports.NumberAttributeConstraintsType$ = exports.NotifyEmailType$ = exports.NotifyConfigurationType$ = exports.NewDeviceMetadataType$ = exports.MFAOptionType$ = exports.MessageTemplateType$ = exports.ManagedLoginBrandingType$ = exports.LogDeliveryConfigurationType$ = exports.LogConfigurationType$ = exports.ListWebAuthnCredentialsResponse$ = exports.ListWebAuthnCredentialsRequest$ = exports.ListUsersResponse$ = exports.ListUsersRequest$ = exports.ListUsersInGroupResponse$ = exports.ListUsersInGroupRequest$ = exports.ListUserPoolsResponse$ = exports.ListUserPoolsRequest$ = exports.ListUserPoolClientsResponse$ = exports.ListUserPoolClientsRequest$ = exports.ListUserImportJobsResponse$ = exports.ListUserImportJobsRequest$ = exports.ListTermsResponse$ = void 0;
exports.UserImportJobType$ = exports.UserContextDataType$ = exports.UserAttributeUpdateSettingsType$ = exports.UpdateUserPoolResponse$ = exports.UpdateUserPoolRequest$ = exports.UpdateUserPoolDomainResponse$ = exports.UpdateUserPoolDomainRequest$ = exports.UpdateUserPoolClientResponse$ = exports.UpdateUserPoolClientRequest$ = exports.UpdateUserAttributesResponse$ = exports.UpdateUserAttributesRequest$ = exports.UpdateTermsResponse$ = exports.UpdateTermsRequest$ = exports.UpdateResourceServerResponse$ = exports.UpdateResourceServerRequest$ = exports.UpdateManagedLoginBrandingResponse$ = exports.UpdateManagedLoginBrandingRequest$ = exports.UpdateIdentityProviderResponse$ = exports.UpdateIdentityProviderRequest$ = exports.UpdateGroupResponse$ = exports.UpdateGroupRequest$ = exports.UpdateDeviceStatusResponse$ = exports.UpdateDeviceStatusRequest$ = exports.UpdateAuthEventFeedbackResponse$ = exports.UpdateAuthEventFeedbackRequest$ = exports.UntagResourceResponse$ = exports.UntagResourceRequest$ = exports.UICustomizationType$ = exports.TokenValidityUnitsType$ = exports.TermsType$ = exports.TermsDescriptionType$ = exports.TagResourceResponse$ = exports.TagResourceRequest$ = exports.StringAttributeConstraintsType$ = exports.StopUserImportJobResponse$ = exports.StopUserImportJobRequest$ = exports.StartWebAuthnRegistrationResponse$ = exports.StartWebAuthnRegistrationRequest$ = exports.StartUserImportJobResponse$ = exports.StartUserImportJobRequest$ = exports.SoftwareTokenMfaSettingsType$ = exports.SoftwareTokenMfaConfigType$ = exports.SMSMfaSettingsType$ = exports.SmsMfaConfigType$ = exports.SmsConfigurationType$ = exports.SignUpResponse$ = exports.SignUpRequest$ = exports.SignInPolicyType$ = exports.SetUserSettingsResponse$ = exports.SetUserSettingsRequest$ = void 0;
exports.CreateIdentityProvider$ = exports.CreateGroup$ = exports.ConfirmSignUp$ = exports.ConfirmForgotPassword$ = exports.ConfirmDevice$ = exports.CompleteWebAuthnRegistration$ = exports.ChangePassword$ = exports.AssociateSoftwareToken$ = exports.AdminUserGlobalSignOut$ = exports.AdminUpdateUserAttributes$ = exports.AdminUpdateDeviceStatus$ = exports.AdminUpdateAuthEventFeedback$ = exports.AdminSetUserSettings$ = exports.AdminSetUserPassword$ = exports.AdminSetUserMFAPreference$ = exports.AdminRespondToAuthChallenge$ = exports.AdminResetUserPassword$ = exports.AdminRemoveUserFromGroup$ = exports.AdminListUserAuthEvents$ = exports.AdminListGroupsForUser$ = exports.AdminListDevices$ = exports.AdminLinkProviderForUser$ = exports.AdminInitiateAuth$ = exports.AdminGetUser$ = exports.AdminGetDevice$ = exports.AdminForgetDevice$ = exports.AdminEnableUser$ = exports.AdminDisableUser$ = exports.AdminDisableProviderForUser$ = exports.AdminDeleteUserAttributes$ = exports.AdminDeleteUser$ = exports.AdminCreateUser$ = exports.AdminConfirmSignUp$ = exports.AdminAddUserToGroup$ = exports.AddCustomAttributes$ = exports.WebAuthnCredentialDescription$ = exports.WebAuthnConfigurationType$ = exports.VerifyUserAttributeResponse$ = exports.VerifyUserAttributeRequest$ = exports.VerifySoftwareTokenResponse$ = exports.VerifySoftwareTokenRequest$ = exports.VerificationMessageTemplateType$ = exports.UserType$ = exports.UserPoolType$ = exports.UserPoolPolicyType$ = exports.UserPoolDescriptionType$ = exports.UserPoolClientType$ = exports.UserPoolClientDescription$ = exports.UserPoolAddOnsType$ = exports.UsernameConfigurationType$ = void 0;
exports.ListTerms$ = exports.ListTagsForResource$ = exports.ListResourceServers$ = exports.ListIdentityProviders$ = exports.ListGroups$ = exports.ListDevices$ = exports.InitiateAuth$ = exports.GlobalSignOut$ = exports.GetUserPoolMfaConfig$ = exports.GetUserAuthFactors$ = exports.GetUserAttributeVerificationCode$ = exports.GetUser$ = exports.GetUICustomization$ = exports.GetTokensFromRefreshToken$ = exports.GetSigningCertificate$ = exports.GetLogDeliveryConfiguration$ = exports.GetIdentityProviderByIdentifier$ = exports.GetGroup$ = exports.GetDevice$ = exports.GetCSVHeader$ = exports.ForgotPassword$ = exports.ForgetDevice$ = exports.DescribeUserPoolDomain$ = exports.DescribeUserPoolClient$ = exports.DescribeUserPool$ = exports.DescribeUserImportJob$ = exports.DescribeTerms$ = exports.DescribeRiskConfiguration$ = exports.DescribeResourceServer$ = exports.DescribeManagedLoginBrandingByClient$ = exports.DescribeManagedLoginBranding$ = exports.DescribeIdentityProvider$ = exports.DeleteWebAuthnCredential$ = exports.DeleteUserPoolDomain$ = exports.DeleteUserPoolClient$ = exports.DeleteUserPool$ = exports.DeleteUserAttributes$ = exports.DeleteUser$ = exports.DeleteTerms$ = exports.DeleteResourceServer$ = exports.DeleteManagedLoginBranding$ = exports.DeleteIdentityProvider$ = exports.DeleteGroup$ = exports.CreateUserPoolDomain$ = exports.CreateUserPoolClient$ = exports.CreateUserPool$ = exports.CreateUserImportJob$ = exports.CreateTerms$ = exports.CreateResourceServer$ = exports.CreateManagedLoginBranding$ = void 0;
exports.VerifyUserAttribute$ = exports.VerifySoftwareToken$ = exports.UpdateUserPoolDomain$ = exports.UpdateUserPoolClient$ = exports.UpdateUserPool$ = exports.UpdateUserAttributes$ = exports.UpdateTerms$ = exports.UpdateResourceServer$ = exports.UpdateManagedLoginBranding$ = exports.UpdateIdentityProvider$ = exports.UpdateGroup$ = exports.UpdateDeviceStatus$ = exports.UpdateAuthEventFeedback$ = exports.UntagResource$ = exports.TagResource$ = exports.StopUserImportJob$ = exports.StartWebAuthnRegistration$ = exports.StartUserImportJob$ = exports.SignUp$ = exports.SetUserSettings$ = exports.SetUserPoolMfaConfig$ = exports.SetUserMFAPreference$ = exports.SetUICustomization$ = exports.SetRiskConfiguration$ = exports.SetLogDeliveryConfiguration$ = exports.RevokeToken$ = exports.RespondToAuthChallenge$ = exports.ResendConfirmationCode$ = exports.ListWebAuthnCredentials$ = exports.ListUsersInGroup$ = exports.ListUsers$ = exports.ListUserPools$ = exports.ListUserPoolClients$ = exports.ListUserImportJobs$ = void 0;
const _A = "Actions";
const _AA = "ApplicationArn";
const _AACUO = "AllowAdminCreateUserOnly";
const _AAUTG = "AdminAddUserToGroup";
const _AAUTGR = "AdminAddUserToGroupRequest";
const _AAl = "AliasAttributes";
const _AAu = "AuthenticatorAttachment";
const _AC = "AvailableChallenges";
const _ACA = "AddCustomAttributes";
const _ACAR = "AddCustomAttributesRequest";
const _ACARd = "AddCustomAttributesResponse";
const _ACSU = "AdminConfirmSignUp";
const _ACSUR = "AdminConfirmSignUpRequest";
const _ACSURd = "AdminConfirmSignUpResponse";
const _ACT = "AnalyticsConfigurationType";
const _ACU = "AdminCreateUser";
const _ACUC = "AdminCreateUserConfig";
const _ACUCT = "AdminCreateUserConfigType";
const _ACUR = "AdminCreateUserRequest";
const _ACURd = "AdminCreateUserResponse";
const _ACn = "AnalyticsConfiguration";
const _ADPFU = "AdminDisableProviderForUser";
const _ADPFUR = "AdminDisableProviderForUserRequest";
const _ADPFURd = "AdminDisableProviderForUserResponse";
const _ADT = "AttributeDataType";
const _ADU = "AdminDeleteUser";
const _ADUA = "AdminDeleteUserAttributes";
const _ADUAR = "AdminDeleteUserAttributesRequest";
const _ADUARd = "AdminDeleteUserAttributesResponse";
const _ADUR = "AdminDeleteUserRequest";
const _ADURd = "AdminDisableUserRequest";
const _ADURdm = "AdminDisableUserResponse";
const _ADUd = "AdminDisableUser";
const _AE = "AuthEvents";
const _AEE = "AliasExistsException";
const _AEI = "AnalyticsEndpointId";
const _AET = "AuthEventType";
const _AETu = "AuthEventsType";
const _AEU = "AdminEnableUser";
const _AEUR = "AdminEnableUserRequest";
const _AEURd = "AdminEnableUserResponse";
const _AF = "AuthFlow";
const _AFAF = "AllowedFirstAuthFactors";
const _AFD = "AdminForgetDevice";
const _AFDR = "AdminForgetDeviceRequest";
const _AGD = "AdminGetDevice";
const _AGDR = "AdminGetDeviceRequest";
const _AGDRd = "AdminGetDeviceResponse";
const _AGU = "AdminGetUser";
const _AGUR = "AdminGetUserRequest";
const _AGURd = "AdminGetUserResponse";
const _AI = "ApplicationId";
const _AIA = "AdminInitiateAuth";
const _AIAR = "AdminInitiateAuthRequest";
const _AIARd = "AdminInitiateAuthResponse";
const _ALD = "AdminListDevices";
const _ALDR = "AdminListDevicesRequest";
const _ALDRd = "AdminListDevicesResponse";
const _ALGFU = "AdminListGroupsForUser";
const _ALGFUR = "AdminListGroupsForUserRequest";
const _ALGFURd = "AdminListGroupsForUserResponse";
const _ALPFU = "AdminLinkProviderForUser";
const _ALPFUR = "AdminLinkProviderForUserRequest";
const _ALPFURd = "AdminLinkProviderForUserResponse";
const _ALT = "AssetListType";
const _ALTt = "AttributeListType";
const _ALUAE = "AdminListUserAuthEvents";
const _ALUAER = "AdminListUserAuthEventsRequest";
const _ALUAERd = "AdminListUserAuthEventsResponse";
const _AM = "AnalyticsMetadata";
const _AMT = "AnalyticsMetadataType";
const _AMt = "AttributeMapping";
const _AN = "AttributeName";
const _AOAF = "AllowedOAuthFlows";
const _AOAFUPC = "AllowedOAuthFlowsUserPoolClient";
const _AOAS = "AllowedOAuthScopes";
const _AP = "AuthParameters";
const _APT = "AuthParametersType";
const _AR = "AuthenticationResult";
const _ARS = "AccountRecoverySetting";
const _ARST = "AccountRecoverySettingType";
const _ART = "AuthenticationResultType";
const _ARTAC = "AdminRespondToAuthChallenge";
const _ARTACR = "AdminRespondToAuthChallengeRequest";
const _ARTACRd = "AdminRespondToAuthChallengeResponse";
const _ARUFG = "AdminRemoveUserFromGroup";
const _ARUFGR = "AdminRemoveUserFromGroupRequest";
const _ARUP = "AdminResetUserPassword";
const _ARUPR = "AdminResetUserPasswordRequest";
const _ARUPRd = "AdminResetUserPasswordResponse";
const _ARVBU = "AttributesRequireVerificationBeforeUpdate";
const _ASAF = "AdvancedSecurityAdditionalFlows";
const _ASAFT = "AdvancedSecurityAdditionalFlowsType";
const _ASM = "AdvancedSecurityMode";
const _AST = "AssociateSoftwareToken";
const _ASTR = "AssociateSoftwareTokenRequest";
const _ASTRs = "AssociateSoftwareTokenResponse";
const _ASUMFAP = "AdminSetUserMFAPreference";
const _ASUMFAPR = "AdminSetUserMFAPreferenceRequest";
const _ASUMFAPRd = "AdminSetUserMFAPreferenceResponse";
const _ASUP = "AdminSetUserPassword";
const _ASUPR = "AdminSetUserPasswordRequest";
const _ASUPRd = "AdminSetUserPasswordResponse";
const _ASUS = "AdminSetUserSettings";
const _ASUSR = "AdminSetUserSettingsRequest";
const _ASUSRd = "AdminSetUserSettingsResponse";
const _ASV = "AuthSessionValidity";
const _AT = "AssetType";
const _ATAT = "AccountTakeoverActionsType";
const _ATATc = "AccountTakeoverActionType";
const _ATG = "AttributesToGet";
const _ATRC = "AccountTakeoverRiskConfiguration";
const _ATRCT = "AccountTakeoverRiskConfigurationType";
const _ATV = "AccessTokenValidity";
const _ATc = "AccessToken";
const _ATt = "AttributeType";
const _ATu = "AuthenticatorTransports";
const _AUAEF = "AdminUpdateAuthEventFeedback";
const _AUAEFR = "AdminUpdateAuthEventFeedbackRequest";
const _AUAEFRd = "AdminUpdateAuthEventFeedbackResponse";
const _AUDS = "AdminUpdateDeviceStatus";
const _AUDSR = "AdminUpdateDeviceStatusRequest";
const _AUDSRd = "AdminUpdateDeviceStatusResponse";
const _AUGSO = "AdminUserGlobalSignOut";
const _AUGSOR = "AdminUserGlobalSignOutRequest";
const _AUGSORd = "AdminUserGlobalSignOutResponse";
const _AUUA = "AdminUpdateUserAttributes";
const _AUUAR = "AdminUpdateUserAttributesRequest";
const _AUUARd = "AdminUpdateUserAttributesResponse";
const _AVA = "AutoVerifiedAttributes";
const _AVT = "AttributeValueType";
const _AWSAI = "AWSAccountId";
const _Ar = "Arn";
const _As = "Assets";
const _At = "Attributes";
const _B = "Bytes";
const _BA = "BucketArn";
const _BE = "BlockEmail";
const _BIPRL = "BlockedIPRangeList";
const _C = "Category";
const _CA = "CustomAttributes";
const _CAC = "CreateAuthChallenge";
const _CALT = "CustomAttributesListType";
const _CAM = "CustomAuthMode";
const _CAe = "CertificateArn";
const _CAr = "CreatedAt";
const _CC = "ConfirmationCode";
const _CCAT = "CompromisedCredentialsActionsType";
const _CCD = "CompromisedCredentialsDetected";
const _CCO = "CredentialCreationOptions";
const _CCRC = "CompromisedCredentialsRiskConfiguration";
const _CCRCT = "CompromisedCredentialsRiskConfigurationType";
const _CD = "ContextData";
const _CDC = "CustomDomainConfig";
const _CDCT = "CustomDomainConfigType";
const _CDD = "CodeDeliveryDetails";
const _CDDL = "CodeDeliveryDetailsList";
const _CDDLT = "CodeDeliveryDetailsListType";
const _CDDT = "CodeDeliveryDetailsType";
const _CDFE = "CodeDeliveryFailureException";
const _CDR = "ConfirmDeviceRequest";
const _CDRo = "ConfirmDeviceResponse";
const _CDT = "ContextDataType";
const _CDo = "CompletionDate";
const _CDon = "ConfirmDevice";
const _CDr = "CreationDate";
const _CDu = "CustomDomain";
const _CELVCT = "CustomEmailLambdaVersionConfigType";
const _CES = "CustomEmailSender";
const _CFD = "CloudFrontDomain";
const _CFDl = "CloudFrontDistribution";
const _CFP = "ConfirmForgotPassword";
const _CFPR = "ConfirmForgotPasswordRequest";
const _CFPRo = "ConfirmForgotPasswordResponse";
const _CG = "CreateGroup";
const _CGR = "CreateGroupRequest";
const _CGRr = "CreateGroupResponse";
const _CI = "ClientId";
const _CIP = "CreateIdentityProvider";
const _CIPR = "CreateIdentityProviderRequest";
const _CIPRr = "CreateIdentityProviderResponse";
const _CIT = "ClientIdType";
const _CIr = "CredentialId";
const _CM = "ClientMetadata";
const _CME = "CodeMismatchException";
const _CMEo = "ConcurrentModificationException";
const _CMLB = "CreateManagedLoginBranding";
const _CMLBR = "CreateManagedLoginBrandingRequest";
const _CMLBRr = "CreateManagedLoginBrandingResponse";
const _CMo = "ColorMode";
const _CMom = "CompletionMessage";
const _CMu = "CustomMessage";
const _CN = "ChallengeName";
const _CNl = "ClientName";
const _CP = "ChallengeParameters";
const _CPR = "ChangePasswordRequest";
const _CPRh = "ChangePasswordResponse";
const _CPh = "ChangePassword";
const _CR = "ChallengeResponses";
const _CRLT = "ChallengeResponseListType";
const _CROND = "ChallengeRequiredOnNewDevice";
const _CRS = "CreateResourceServer";
const _CRSR = "CreateResourceServerRequest";
const _CRSRr = "CreateResourceServerResponse";
const _CRT = "ChallengeResponseType";
const _CRTh = "ChallengeResponsesType";
const _CRh = "ChallengeResponse";
const _CS = "ConfigurationSet";
const _CSMSLVCT = "CustomSMSLambdaVersionConfigType";
const _CSMSS = "CustomSMSSender";
const _CSS = "CSS";
const _CSSV = "CSSVersion";
const _CST = "ClientSecretType";
const _CSU = "ConfirmSignUp";
const _CSUR = "ConfirmSignUpRequest";
const _CSURo = "ConfirmSignUpResponse";
const _CSVH = "CSVHeader";
const _CSa = "CaseSensitive";
const _CSl = "ClientSecret";
const _CT = "CreateTerms";
const _CTR = "CreateTermsRequest";
const _CTRr = "CreateTermsResponse";
const _CUAF = "ConfiguredUserAuthFactors";
const _CUIJ = "CreateUserImportJob";
const _CUIJR = "CreateUserImportJobRequest";
const _CUIJRr = "CreateUserImportJobResponse";
const _CUP = "CreateUserPool";
const _CUPC = "CreateUserPoolClient";
const _CUPCR = "CreateUserPoolClientRequest";
const _CUPCRr = "CreateUserPoolClientResponse";
const _CUPD = "CreateUserPoolDomain";
const _CUPDR = "CreateUserPoolDomainRequest";
const _CUPDRr = "CreateUserPoolDomainResponse";
const _CUPR = "CreateUserPoolRequest";
const _CUPRr = "CreateUserPoolResponse";
const _CURL = "CallbackURLs";
const _CWAR = "CompleteWebAuthnRegistration";
const _CWARR = "CompleteWebAuthnRegistrationRequest";
const _CWARRo = "CompleteWebAuthnRegistrationResponse";
const _CWLC = "CloudWatchLogsConfiguration";
const _CWLCT = "CloudWatchLogsConfigurationType";
const _CWLRA = "CloudWatchLogsRoleArn";
const _Ce = "Certificate";
const _Ci = "City";
const _Co = "Country";
const _Cod = "Code";
const _Cr = "Credential";
const _Cre = "Credentials";
const _D = "Device";
const _DA = "DeviceAttributes";
const _DAC = "DefineAuthChallenge";
const _DC = "DeviceConfiguration";
const _DCD = "DeviceCreateDate";
const _DCT = "DeviceConfigurationType";
const _DD = "DomainDescription";
const _DDM = "DesiredDeliveryMediums";
const _DDT = "DomainDescriptionType";
const _DEO = "DefaultEmailOption";
const _DG = "DeleteGroup";
const _DGK = "DeviceGroupKey";
const _DGR = "DeleteGroupRequest";
const _DIP = "DeleteIdentityProvider";
const _DIPR = "DeleteIdentityProviderRequest";
const _DIPRe = "DescribeIdentityProviderRequest";
const _DIPRes = "DescribeIdentityProviderResponse";
const _DIPe = "DescribeIdentityProvider";
const _DK = "DeviceKey";
const _DKEE = "DeviceKeyExistsException";
const _DLAD = "DeviceLastAuthenticatedDate";
const _DLMD = "DeviceLastModifiedDate";
const _DLT = "DeviceListType";
const _DM = "DeliveryMedium";
const _DMLB = "DeleteManagedLoginBranding";
const _DMLBBC = "DescribeManagedLoginBrandingByClient";
const _DMLBBCR = "DescribeManagedLoginBrandingByClientRequest";
const _DMLBBCRe = "DescribeManagedLoginBrandingByClientResponse";
const _DMLBR = "DeleteManagedLoginBrandingRequest";
const _DMLBRe = "DescribeManagedLoginBrandingRequest";
const _DMLBRes = "DescribeManagedLoginBrandingResponse";
const _DMLBe = "DescribeManagedLoginBranding";
const _DN = "DeviceName";
const _DOA = "DeveloperOnlyAttribute";
const _DOROUP = "DeviceOnlyRememberedOnUserPrompt";
const _DP = "DeletionProtection";
const _DPE = "DuplicateProviderException";
const _DRC = "DescribeRiskConfiguration";
const _DRCR = "DescribeRiskConfigurationRequest";
const _DRCRe = "DescribeRiskConfigurationResponse";
const _DRS = "DeviceRememberedStatus";
const _DRSR = "DeleteResourceServerRequest";
const _DRSRe = "DescribeResourceServerRequest";
const _DRSRes = "DescribeResourceServerResponse";
const _DRSe = "DeleteResourceServer";
const _DRSes = "DescribeResourceServer";
const _DRURI = "DefaultRedirectURI";
const _DSVC = "DeviceSecretVerifierConfig";
const _DSVCT = "DeviceSecretVerifierConfigType";
const _DT = "DeviceType";
const _DTR = "DeleteTermsRequest";
const _DTRe = "DescribeTermsRequest";
const _DTRes = "DescribeTermsResponse";
const _DTe = "DeleteTerms";
const _DTes = "DescribeTerms";
const _DU = "DestinationUser";
const _DUA = "DeleteUserAttributes";
const _DUAR = "DeleteUserAttributesRequest";
const _DUARe = "DeleteUserAttributesResponse";
const _DUIJ = "DescribeUserImportJob";
const _DUIJR = "DescribeUserImportJobRequest";
const _DUIJRe = "DescribeUserImportJobResponse";
const _DUP = "DeleteUserPool";
const _DUPC = "DeleteUserPoolClient";
const _DUPCR = "DeleteUserPoolClientRequest";
const _DUPCRe = "DescribeUserPoolClientRequest";
const _DUPCRes = "DescribeUserPoolClientResponse";
const _DUPCe = "DescribeUserPoolClient";
const _DUPD = "DeleteUserPoolDomain";
const _DUPDR = "DeleteUserPoolDomainRequest";
const _DUPDRe = "DeleteUserPoolDomainResponse";
const _DUPDRes = "DescribeUserPoolDomainRequest";
const _DUPDResc = "DescribeUserPoolDomainResponse";
const _DUPDe = "DescribeUserPoolDomain";
const _DUPR = "DeleteUserPoolRequest";
const _DUPRe = "DescribeUserPoolRequest";
const _DUPRes = "DescribeUserPoolResponse";
const _DUPe = "DescribeUserPool";
const _DUR = "DeleteUserRequest";
const _DUe = "DeleteUser";
const _DWAC = "DeleteWebAuthnCredential";
const _DWACR = "DeleteWebAuthnCredentialRequest";
const _DWACRe = "DeleteWebAuthnCredentialResponse";
const _De = "Devices";
const _Des = "Destination";
const _Desc = "Description";
const _Do = "Domain";
const _E = "Enabled";
const _EA = "EventAction";
const _EAF = "ExplicitAuthFlows";
const _EC = "EmailConfiguration";
const _ECD = "EventContextData";
const _ECDT = "EventContextDataType";
const _ECE = "ExpiredCodeException";
const _ECF = "EmailConfigurationFailure";
const _ECT = "EmailConfigurationType";
const _ED = "EncodedData";
const _EF = "EventFeedback";
const _EFT = "EventFeedbackType";
const _EFv = "EventFilter";
const _EI = "EventId";
const _EIx = "ExternalId";
const _EIxp = "ExpiresIn";
const _EM = "EmailMessage";
const _EMBL = "EmailMessageByLink";
const _EMC = "EmailMfaConfiguration";
const _EMCT = "EmailMfaConfigType";
const _EMS = "EmailMfaSettings";
const _EMST = "EmailMfaSettingsType";
const _ENOU = "EstimatedNumberOfUsers";
const _EPAUCD = "EnablePropagateAdditionalUserContextData";
const _ER = "EventResponse";
const _ERT = "EventRiskType";
const _ERv = "EventRisk";
const _ES = "EventSource";
const _ESA = "EmailSendingAccount";
const _ESBL = "EmailSubjectByLink";
const _ESTMFAE = "EnableSoftwareTokenMFAException";
const _ESm = "EmailSubject";
const _ET = "EventType";
const _ETR = "EnableTokenRevocation";
const _EVM = "EmailVerificationMessage";
const _EVS = "EmailVerificationSubject";
const _En = "Enforcement";
const _Ex = "Extension";
const _F = "From";
const _FAC = "ForceAliasCreation";
const _FC = "FirehoseConfiguration";
const _FCN = "FriendlyCredentialName";
const _FCT = "FirehoseConfigurationType";
const _FD = "FeedbackDate";
const _FDN = "FriendlyDeviceName";
const _FDR = "ForgetDeviceRequest";
const _FDo = "ForgetDevice";
const _FE = "ForbiddenException";
const _FP = "ForgotPassword";
const _FPR = "ForgotPasswordRequest";
const _FPRo = "ForgotPasswordResponse";
const _FT = "FeedbackToken";
const _FU = "FailedUsers";
const _FUITE = "FeatureUnavailableInTierException";
const _FV = "FeedbackValue";
const _Fe = "Feature";
const _Fi = "Filter";
const _G = "Groups";
const _GCSVH = "GetCSVHeader";
const _GCSVHR = "GetCSVHeaderRequest";
const _GCSVHRe = "GetCSVHeaderResponse";
const _GD = "GetDevice";
const _GDR = "GetDeviceRequest";
const _GDRe = "GetDeviceResponse";
const _GEE = "GroupExistsException";
const _GG = "GetGroup";
const _GGR = "GetGroupRequest";
const _GGRe = "GetGroupResponse";
const _GIPBI = "GetIdentityProviderByIdentifier";
const _GIPBIR = "GetIdentityProviderByIdentifierRequest";
const _GIPBIRe = "GetIdentityProviderByIdentifierResponse";
const _GLDC = "GetLogDeliveryConfiguration";
const _GLDCR = "GetLogDeliveryConfigurationRequest";
const _GLDCRe = "GetLogDeliveryConfigurationResponse";
const _GLT = "GroupListType";
const _GN = "GroupName";
const _GS = "GenerateSecret";
const _GSC = "GetSigningCertificate";
const _GSCR = "GetSigningCertificateRequest";
const _GSCRe = "GetSigningCertificateResponse";
const _GSO = "GlobalSignOut";
const _GSOR = "GlobalSignOutRequest";
const _GSORl = "GlobalSignOutResponse";
const _GT = "GroupType";
const _GTFRT = "GetTokensFromRefreshToken";
const _GTFRTR = "GetTokensFromRefreshTokenRequest";
const _GTFRTRe = "GetTokensFromRefreshTokenResponse";
const _GU = "GetUser";
const _GUAF = "GetUserAuthFactors";
const _GUAFR = "GetUserAuthFactorsRequest";
const _GUAFRe = "GetUserAuthFactorsResponse";
const _GUAVC = "GetUserAttributeVerificationCode";
const _GUAVCR = "GetUserAttributeVerificationCodeRequest";
const _GUAVCRe = "GetUserAttributeVerificationCodeResponse";
const _GUIC = "GetUICustomization";
const _GUICR = "GetUICustomizationRequest";
const _GUICRe = "GetUICustomizationResponse";
const _GUPMC = "GetUserPoolMfaConfig";
const _GUPMCR = "GetUserPoolMfaConfigRequest";
const _GUPMCRe = "GetUserPoolMfaConfigResponse";
const _GUR = "GetUserRequest";
const _GURe = "GetUserResponse";
const _Gr = "Group";
const _HA = "HighAction";
const _HB = "HtmlBody";
const _HH = "HttpHeaders";
const _HHL = "HttpHeaderList";
const _HHt = "HttpHeader";
const _I = "Identifier";
const _IA = "IpAddress";
const _IAR = "InitiateAuthRequest";
const _IARn = "InitiateAuthResponse";
const _IAn = "InitiateAuth";
const _IEE = "InternalErrorException";
const _IERAPE = "InvalidEmailRoleAccessPolicyException";
const _IF = "InboundFederation";
const _IFLT = "InboundFederationLambdaType";
const _IFm = "ImageFile";
const _II = "IdpIdentifiers";
const _IId = "IdpIdentifier";
const _ILRE = "InvalidLambdaResponseException";
const _IMT = "InviteMessageTemplate";
const _IOAFE = "InvalidOAuthFlowException";
const _IP = "IdentityProvider";
const _IPE = "InvalidParameterException";
const _IPEn = "InvalidPasswordException";
const _IPT = "IdentityProviderType";
const _ISRAPE = "InvalidSmsRoleAccessPolicyException";
const _ISRTRE = "InvalidSmsRoleTrustRelationshipException";
const _IT = "IdToken";
const _ITV = "IdTokenValidity";
const _IU = "ImageUrl";
const _IUPCE = "InvalidUserPoolConfigurationException";
const _IUm = "ImportedUsers";
const _Id = "Id";
const _JI = "JobId";
const _JN = "JobName";
const _KMSKID = "KMSKeyID";
const _L = "Limit";
const _LA = "LowAction";
const _LAa = "LambdaArn";
const _LC = "LambdaConfig";
const _LCLT = "LogConfigurationListType";
const _LCT = "LambdaConfigType";
const _LCTo = "LogConfigurationType";
const _LCo = "LogConfigurations";
const _LD = "ListDevices";
const _LDC = "LogDeliveryConfiguration";
const _LDCT = "LogDeliveryConfigurationType";
const _LDR = "ListDevicesRequest";
const _LDRi = "ListDevicesResponse";
const _LEE = "LimitExceededException";
const _LG = "ListGroups";
const _LGA = "LogGroupArn";
const _LGR = "ListGroupsRequest";
const _LGRi = "ListGroupsResponse";
const _LIP = "ListIdentityProviders";
const _LIPR = "ListIdentityProvidersRequest";
const _LIPRi = "ListIdentityProvidersResponse";
const _LL = "LogLevel";
const _LMD = "LastModifiedDate";
const _LRS = "ListResourceServers";
const _LRSR = "ListResourceServersRequest";
const _LRSRi = "ListResourceServersResponse";
const _LT = "ListTerms";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _LTR = "ListTermsRequest";
const _LTRi = "ListTermsResponse";
const _LU = "ListUsers";
const _LUIG = "ListUsersInGroup";
const _LUIGR = "ListUsersInGroupRequest";
const _LUIGRi = "ListUsersInGroupResponse";
const _LUIJ = "ListUserImportJobs";
const _LUIJR = "ListUserImportJobsRequest";
const _LUIJRi = "ListUserImportJobsResponse";
const _LUP = "ListUserPools";
const _LUPC = "ListUserPoolClients";
const _LUPCR = "ListUserPoolClientsRequest";
const _LUPCRi = "ListUserPoolClientsResponse";
const _LUPR = "ListUserPoolsRequest";
const _LUPRi = "ListUserPoolsResponse";
const _LUR = "ListUsersRequest";
const _LURL = "LogoutURLs";
const _LURi = "ListUsersResponse";
const _LV = "LambdaVersion";
const _LWAC = "ListWebAuthnCredentials";
const _LWACR = "ListWebAuthnCredentialsRequest";
const _LWACRi = "ListWebAuthnCredentialsResponse";
const _Li = "Links";
const _M = "Message";
const _MA = "MediumAction";
const _MAe = "MessageAction";
const _MC = "MfaConfiguration";
const _ME = "MfaEmail";
const _MFAMNFE = "MFAMethodNotFoundException";
const _MFAO = "MFAOptions";
const _MFAOLT = "MFAOptionListType";
const _MFAOT = "MFAOptionType";
const _ML = "MinimumLength";
const _MLB = "ManagedLoginBranding";
const _MLBEE = "ManagedLoginBrandingExistsException";
const _MLBI = "ManagedLoginBrandingId";
const _MLBT = "ManagedLoginBrandingType";
const _MLV = "ManagedLoginVersion";
const _MLa = "MaxLength";
const _MLi = "MinLength";
const _MR = "MaxResults";
const _MTT = "MessageTemplateType";
const _MV = "MinValue";
const _MVa = "MaxValue";
const _Mu = "Mutable";
const _N = "Notify";
const _NAC = "NumberAttributeConstraints";
const _NACT = "NumberAttributeConstraintsType";
const _NAE = "NotAuthorizedException";
const _NAEo = "NoActionEmail";
const _NC = "NotifyConfiguration";
const _NCT = "NotifyConfigurationType";
const _NDM = "NewDeviceMetadata";
const _NDMT = "NewDeviceMetadataType";
const _NET = "NotifyEmailType";
const _NT = "NextToken";
const _Na = "Name";
const _P = "Password";
const _PA = "PreAuthentication";
const _PAN = "ProviderAttributeName";
const _PAV = "ProviderAttributeValue";
const _PAo = "PostAuthentication";
const _PC = "PostConfirmation";
const _PD = "ProviderDetails";
const _PDr = "ProviderDescription";
const _PHPVE = "PasswordHistoryPolicyViolationException";
const _PHS = "PasswordHistorySize";
const _PLT = "ProvidersListType";
const _PM = "PreferredMfa";
const _PMS = "PreferredMfaSetting";
const _PN = "ProviderName";
const _PNME = "PreconditionNotMetException";
const _PNo = "PoolName";
const _PP = "ProposedPassword";
const _PPT = "PasswordPolicyType";
const _PPa = "PasswordPolicy";
const _PPr = "PreviousPassword";
const _PRRE = "PasswordResetRequiredException";
const _PSU = "PreSignUp";
const _PSUr = "PreSignedUrl";
const _PT = "PasswordType";
const _PTG = "PreTokenGeneration";
const _PTGC = "PreTokenGenerationConfig";
const _PTGVCT = "PreTokenGenerationVersionConfigType";
const _PTa = "PaginationToken";
const _PTr = "ProviderType";
const _PUEE = "PreventUserExistenceErrors";
const _PUIT = "ProviderUserIdentifierType";
const _PV = "PasswordVerifier";
const _Pe = "Permanent";
const _Po = "Policies";
const _Pr = "Precedence";
const _Pri = "Priority";
const _Pro = "Provider";
const _Prov = "Providers";
const _R = "Required";
const _RA = "RoleArn";
const _RAe = "ReadAttributes";
const _RAes = "ResourceArn";
const _RC = "RiskConfiguration";
const _RCC = "ResendConfirmationCode";
const _RCCR = "ResendConfirmationCodeRequest";
const _RCCRe = "ResendConfirmationCodeResponse";
const _RCT = "RiskConfigurationType";
const _RD = "RiskDecision";
const _REC = "RiskExceptionConfiguration";
const _RECT = "RiskExceptionConfigurationType";
const _RGPS = "RetryGracePeriodSeconds";
const _RI = "ResourceId";
const _RL = "RiskLevel";
const _RLe = "RequireLowercase";
const _RM = "RecoveryMechanisms";
const _RMR = "ReturnMergedResources";
const _RMT = "RecoveryMechanismsType";
const _RN = "RequireNumbers";
const _RNFE = "ResourceNotFoundException";
const _ROT = "RecoveryOptionType";
const _RPI = "RelyingPartyId";
const _RS = "ResourceServer";
const _RSLT = "ResourceServersListType";
const _RSSLT = "ResourceServerScopeListType";
const _RSST = "ResourceServerScopeType";
const _RST = "ResourceServerType";
const _RSe = "ResourceServers";
const _RSeq = "RequireSymbols";
const _RT = "RefreshToken";
const _RTAC = "RespondToAuthChallenge";
const _RTACR = "RespondToAuthChallengeRequest";
const _RTACRe = "RespondToAuthChallengeResponse";
const _RTEA = "ReplyToEmailAddress";
const _RTR = "RefreshTokenRotation";
const _RTRE = "RefreshTokenReuseException";
const _RTRT = "RefreshTokenRotationType";
const _RTRe = "RevokeTokenRequest";
const _RTRev = "RevokeTokenResponse";
const _RTV = "RefreshTokenValidity";
const _RTe = "ReplyTo";
const _RTev = "RevokeToken";
const _RU = "RequireUppercase";
const _S = "Session";
const _SA = "SourceArn";
const _SAC = "StringAttributeConstraints";
const _SACT = "StringAttributeConstraintsType";
const _SALT = "SchemaAttributesListType";
const _SAM = "SmsAuthenticationMessage";
const _SAT = "SchemaAttributeType";
const _SAc = "SchemaAttributes";
const _SAt = "StreamArn";
const _SB = "S3Bucket";
const _SC = "SecretCode";
const _SCA = "SnsCallerArn";
const _SCF = "SmsConfigurationFailure";
const _SCT = "SecretCodeType";
const _SCTm = "SmsConfigurationType";
const _SCTo = "S3ConfigurationType";
const _SCm = "SmsConfiguration";
const _SCo = "S3Configuration";
const _SD = "ScopeDescription";
const _SDNEE = "ScopeDoesNotExistException";
const _SDt = "StartDate";
const _SH = "SecretHash";
const _SHT = "SecretHashType";
const _SIP = "SupportedIdentityProviders";
const _SIPRL = "SkippedIPRangeList";
const _SIPT = "SignInPolicyType";
const _SIPi = "SignInPolicy";
const _SLDC = "SetLogDeliveryConfiguration";
const _SLDCR = "SetLogDeliveryConfigurationRequest";
const _SLDCRe = "SetLogDeliveryConfigurationResponse";
const _SM = "SmsMessage";
const _SMC = "SmsMfaConfiguration";
const _SMCT = "SmsMfaConfigType";
const _SMSM = "SMSMessage";
const _SMSMS = "SMSMfaSettings";
const _SMSMST = "SMSMfaSettingsType";
const _SN = "ServerName";
const _SNc = "ScopeName";
const _SP = "ServerPath";
const _SR = "SnsRegion";
const _SRC = "SetRiskConfiguration";
const _SRCR = "SetRiskConfigurationRequest";
const _SRCRe = "SetRiskConfigurationResponse";
const _ST = "SessionType";
const _STMC = "SoftwareTokenMfaConfiguration";
const _STMCT = "SoftwareTokenMfaConfigType";
const _STMFANFE = "SoftwareTokenMFANotFoundException";
const _STMFAUCT = "SoftwareTokenMFAUserCodeType";
const _STMS = "SoftwareTokenMfaSettings";
const _STMST = "SoftwareTokenMfaSettingsType";
const _SU = "SourceUser";
const _SUIC = "SetUICustomization";
const _SUICR = "SetUICustomizationRequest";
const _SUICRe = "SetUICustomizationResponse";
const _SUIJ = "StartUserImportJob";
const _SUIJR = "StartUserImportJobRequest";
const _SUIJRt = "StartUserImportJobResponse";
const _SUIJRto = "StopUserImportJobRequest";
const _SUIJRtop = "StopUserImportJobResponse";
const _SUIJt = "StopUserImportJob";
const _SUMFAP = "SetUserMFAPreference";
const _SUMFAPR = "SetUserMFAPreferenceRequest";
const _SUMFAPRe = "SetUserMFAPreferenceResponse";
const _SUPMC = "SetUserPoolMfaConfig";
const _SUPMCR = "SetUserPoolMfaConfigRequest";
const _SUPMCRe = "SetUserPoolMfaConfigResponse";
const _SUR = "SignUpRequest";
const _SURi = "SignUpResponse";
const _SUS = "SetUserSettings";
const _SUSR = "SetUserSettingsRequest";
const _SUSRe = "SetUserSettingsResponse";
const _SUi = "SignUp";
const _SUk = "SkippedUsers";
const _SVM = "SmsVerificationMessage";
const _SWAR = "StartWebAuthnRegistration";
const _SWARR = "StartWebAuthnRegistrationRequest";
const _SWARRt = "StartWebAuthnRegistrationResponse";
const _Sa = "Salt";
const _Sc = "Scopes";
const _Sch = "Schema";
const _Se = "Settings";
const _St = "Status";
const _Su = "Subject";
const _T = "Terms";
const _TB = "TextBody";
const _TCNAE = "TierChangeNotAllowedException";
const _TDLT = "TermsDescriptionListType";
const _TDT = "TermsDescriptionType";
const _TEE = "TermsExistsException";
const _TI = "TermsId";
const _TK = "TagKeys";
const _TMFAE = "TooManyFailedAttemptsException";
const _TMRE = "TooManyRequestsException";
const _TMT = "TokenModelType";
const _TN = "TermsName";
const _TP = "TemporaryPassword";
const _TPVD = "TemporaryPasswordValidityDays";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _TS = "TermsSource";
const _TT = "TokenType";
const _TTe = "TermsType";
const _TVU = "TokenValidityUnits";
const _TVUT = "TokenValidityUnitsType";
const _Ta = "Tags";
const _Ti = "Timezone";
const _To = "Token";
const _U = "Username";
const _UA = "UserAttributes";
const _UAEF = "UpdateAuthEventFeedback";
const _UAEFR = "UpdateAuthEventFeedbackRequest";
const _UAEFRp = "UpdateAuthEventFeedbackResponse";
const _UAN = "UserAttributeNames";
const _UAUS = "UserAttributeUpdateSettings";
const _UAUST = "UserAttributeUpdateSettingsType";
const _UAVD = "UnusedAccountValidityDays";
const _UAs = "UsernameAttributes";
const _UC = "UsernameConfiguration";
const _UCD = "UserCreateDate";
const _UCDT = "UserContextDataType";
const _UCDs = "UserContextData";
const _UCN = "UserConfirmationNecessary";
const _UCPV = "UseCognitoProvidedValues";
const _UCT = "UsernameConfigurationType";
const _UCs = "UserConfirmed";
const _UCse = "UserCode";
const _UDS = "UserDataShared";
const _UDSR = "UpdateDeviceStatusRequest";
const _UDSRp = "UpdateDeviceStatusResponse";
const _UDSp = "UpdateDeviceStatus";
const _UE = "UnauthorizedException";
const _UEE = "UsernameExistsException";
const _UG = "UpdateGroup";
const _UGR = "UpdateGroupRequest";
const _UGRp = "UpdateGroupResponse";
const _UIC = "UICustomization";
const _UICT = "UICustomizationType";
const _UIIPE = "UserImportInProgressException";
const _UIJ = "UserImportJob";
const _UIJLT = "UserImportJobsListType";
const _UIJT = "UserImportJobType";
const _UIJs = "UserImportJobs";
const _UIP = "UpdateIdentityProvider";
const _UIPE = "UnsupportedIdentityProviderException";
const _UIPR = "UpdateIdentityProviderRequest";
const _UIPRp = "UpdateIdentityProviderResponse";
const _ULE = "UnexpectedLambdaException";
const _ULMD = "UserLastModifiedDate";
const _ULT = "UsersListType";
const _ULVE = "UserLambdaValidationException";
const _UM = "UserMigration";
const _UMFASL = "UserMFASettingList";
const _UMLB = "UpdateManagedLoginBranding";
const _UMLBR = "UpdateManagedLoginBrandingRequest";
const _UMLBRp = "UpdateManagedLoginBrandingResponse";
const _UNCE = "UserNotConfirmedException";
const _UNFE = "UserNotFoundException";
const _UOE = "UnsupportedOperationException";
const _UP = "UserPool";
const _UPAO = "UserPoolAddOns";
const _UPAONEE = "UserPoolAddOnNotEnabledException";
const _UPAOT = "UserPoolAddOnsType";
const _UPC = "UserPoolClient";
const _UPCD = "UserPoolClientDescription";
const _UPCLT = "UserPoolClientListType";
const _UPCT = "UserPoolClientType";
const _UPCs = "UserPoolClients";
const _UPDT = "UserPoolDescriptionType";
const _UPI = "UserPoolId";
const _UPLT = "UserPoolListType";
const _UPPT = "UserPoolPolicyType";
const _UPT = "UserPoolTags";
const _UPTE = "UserPoolTaggingException";
const _UPTs = "UserPoolTier";
const _UPTse = "UserPoolType";
const _UPs = "UserPools";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _URS = "UpdateResourceServer";
const _URSR = "UpdateResourceServerRequest";
const _URSRp = "UpdateResourceServerResponse";
const _US = "UserStatus";
const _USs = "UserSub";
const _UT = "UsernameType";
const _UTR = "UpdateTermsRequest";
const _UTRp = "UpdateTermsResponse";
const _UTTE = "UnsupportedTokenTypeException";
const _UTp = "UpdateTerms";
const _UTs = "UserType";
const _UUA = "UpdateUserAttributes";
const _UUAR = "UpdateUserAttributesRequest";
const _UUARp = "UpdateUserAttributesResponse";
const _UUP = "UpdateUserPool";
const _UUPC = "UpdateUserPoolClient";
const _UUPCR = "UpdateUserPoolClientRequest";
const _UUPCRp = "UpdateUserPoolClientResponse";
const _UUPD = "UpdateUserPoolDomain";
const _UUPDR = "UpdateUserPoolDomainRequest";
const _UUPDRp = "UpdateUserPoolDomainResponse";
const _UUPR = "UpdateUserPoolRequest";
const _UUPRp = "UpdateUserPoolResponse";
const _UUSE = "UnsupportedUserStateException";
const _UV = "UserVerification";
const _Us = "User";
const _Use = "Users";
const _V = "Value";
const _VACR = "VerifyAuthChallengeResponse";
const _VD = "ValidationData";
const _VMT = "VerificationMessageTemplate";
const _VMTT = "VerificationMessageTemplateType";
const _VST = "VerifySoftwareToken";
const _VSTR = "VerifySoftwareTokenRequest";
const _VSTRe = "VerifySoftwareTokenResponse";
const _VUA = "VerifyUserAttribute";
const _VUAR = "VerifyUserAttributeRequest";
const _VUARe = "VerifyUserAttributeResponse";
const _Ve = "Version";
const _WA = "WriteAttributes";
const _WAC = "WebAuthnConfiguration";
const _WACD = "WebAuthnCredentialDescription";
const _WACDLT = "WebAuthnCredentialDescriptionListType";
const _WACME = "WebAuthnClientMismatchException";
const _WACMEe = "WebAuthnConfigurationMissingException";
const _WACNFE = "WebAuthnChallengeNotFoundException";
const _WACNSE = "WebAuthnCredentialNotSupportedException";
const _WACT = "WebAuthnConfigurationType";
const _WANEE = "WebAuthnNotEnabledException";
const _WAONAE = "WebAuthnOriginNotAllowedException";
const _WARPME = "WebAuthnRelyingPartyMismatchException";
const _c = "client";
const _e = "error";
const _hE = "httpError";
const _hN = "headerName";
const _hV = "headerValue";
const _m = "message";
const _rC = "reasonCode";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.cognitoidentityprovider";
const _se = "server";
const n0 = "com.amazonaws.cognitoidentityprovider";
const schema_1 = require("@smithy/core/schema");
const CognitoIdentityProviderServiceException_1 = require("../models/CognitoIdentityProviderServiceException");
const errors_1 = require("../models/errors");
const _s_registry = schema_1.TypeRegistry.for(_s);
exports.CognitoIdentityProviderServiceException$ = [-3, _s, "CognitoIdentityProviderServiceException", 0, [], []];
_s_registry.registerError(exports.CognitoIdentityProviderServiceException$, CognitoIdentityProviderServiceException_1.CognitoIdentityProviderServiceException);
const n0_registry = schema_1.TypeRegistry.for(n0);
exports.AliasExistsException$ = [-3, n0, _AEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.AliasExistsException$, errors_1.AliasExistsException);
exports.CodeDeliveryFailureException$ = [-3, n0, _CDFE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.CodeDeliveryFailureException$, errors_1.CodeDeliveryFailureException);
exports.CodeMismatchException$ = [-3, n0, _CME,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.CodeMismatchException$, errors_1.CodeMismatchException);
exports.ConcurrentModificationException$ = [-3, n0, _CMEo,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.ConcurrentModificationException$, errors_1.ConcurrentModificationException);
exports.DeviceKeyExistsException$ = [-3, n0, _DKEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.DeviceKeyExistsException$, errors_1.DeviceKeyExistsException);
exports.DuplicateProviderException$ = [-3, n0, _DPE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.DuplicateProviderException$, errors_1.DuplicateProviderException);
exports.EnableSoftwareTokenMFAException$ = [-3, n0, _ESTMFAE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.EnableSoftwareTokenMFAException$, errors_1.EnableSoftwareTokenMFAException);
exports.ExpiredCodeException$ = [-3, n0, _ECE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.ExpiredCodeException$, errors_1.ExpiredCodeException);
exports.FeatureUnavailableInTierException$ = [-3, n0, _FUITE,
    { [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
n0_registry.registerError(exports.FeatureUnavailableInTierException$, errors_1.FeatureUnavailableInTierException);
exports.ForbiddenException$ = [-3, n0, _FE,
    { [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
n0_registry.registerError(exports.ForbiddenException$, errors_1.ForbiddenException);
exports.GroupExistsException$ = [-3, n0, _GEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.GroupExistsException$, errors_1.GroupExistsException);
exports.InternalErrorException$ = [-3, n0, _IEE,
    { [_e]: _se },
    [_m],
    [0]
];
n0_registry.registerError(exports.InternalErrorException$, errors_1.InternalErrorException);
exports.InvalidEmailRoleAccessPolicyException$ = [-3, n0, _IERAPE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.InvalidEmailRoleAccessPolicyException$, errors_1.InvalidEmailRoleAccessPolicyException);
exports.InvalidLambdaResponseException$ = [-3, n0, _ILRE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.InvalidLambdaResponseException$, errors_1.InvalidLambdaResponseException);
exports.InvalidOAuthFlowException$ = [-3, n0, _IOAFE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.InvalidOAuthFlowException$, errors_1.InvalidOAuthFlowException);
exports.InvalidParameterException$ = [-3, n0, _IPE,
    { [_e]: _c, [_hE]: 400 },
    [_m, _rC],
    [0, 0]
];
n0_registry.registerError(exports.InvalidParameterException$, errors_1.InvalidParameterException);
exports.InvalidPasswordException$ = [-3, n0, _IPEn,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.InvalidPasswordException$, errors_1.InvalidPasswordException);
exports.InvalidSmsRoleAccessPolicyException$ = [-3, n0, _ISRAPE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.InvalidSmsRoleAccessPolicyException$, errors_1.InvalidSmsRoleAccessPolicyException);
exports.InvalidSmsRoleTrustRelationshipException$ = [-3, n0, _ISRTRE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.InvalidSmsRoleTrustRelationshipException$, errors_1.InvalidSmsRoleTrustRelationshipException);
exports.InvalidUserPoolConfigurationException$ = [-3, n0, _IUPCE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.InvalidUserPoolConfigurationException$, errors_1.InvalidUserPoolConfigurationException);
exports.LimitExceededException$ = [-3, n0, _LEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.LimitExceededException$, errors_1.LimitExceededException);
exports.ManagedLoginBrandingExistsException$ = [-3, n0, _MLBEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.ManagedLoginBrandingExistsException$, errors_1.ManagedLoginBrandingExistsException);
exports.MFAMethodNotFoundException$ = [-3, n0, _MFAMNFE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.MFAMethodNotFoundException$, errors_1.MFAMethodNotFoundException);
exports.NotAuthorizedException$ = [-3, n0, _NAE,
    { [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
n0_registry.registerError(exports.NotAuthorizedException$, errors_1.NotAuthorizedException);
exports.PasswordHistoryPolicyViolationException$ = [-3, n0, _PHPVE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.PasswordHistoryPolicyViolationException$, errors_1.PasswordHistoryPolicyViolationException);
exports.PasswordResetRequiredException$ = [-3, n0, _PRRE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.PasswordResetRequiredException$, errors_1.PasswordResetRequiredException);
exports.PreconditionNotMetException$ = [-3, n0, _PNME,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.PreconditionNotMetException$, errors_1.PreconditionNotMetException);
exports.RefreshTokenReuseException$ = [-3, n0, _RTRE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.RefreshTokenReuseException$, errors_1.RefreshTokenReuseException);
exports.ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(exports.ResourceNotFoundException$, errors_1.ResourceNotFoundException);
exports.ScopeDoesNotExistException$ = [-3, n0, _SDNEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.ScopeDoesNotExistException$, errors_1.ScopeDoesNotExistException);
exports.SoftwareTokenMFANotFoundException$ = [-3, n0, _STMFANFE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.SoftwareTokenMFANotFoundException$, errors_1.SoftwareTokenMFANotFoundException);
exports.TermsExistsException$ = [-3, n0, _TEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.TermsExistsException$, errors_1.TermsExistsException);
exports.TierChangeNotAllowedException$ = [-3, n0, _TCNAE,
    { [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
n0_registry.registerError(exports.TierChangeNotAllowedException$, errors_1.TierChangeNotAllowedException);
exports.TooManyFailedAttemptsException$ = [-3, n0, _TMFAE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.TooManyFailedAttemptsException$, errors_1.TooManyFailedAttemptsException);
exports.TooManyRequestsException$ = [-3, n0, _TMRE,
    { [_e]: _c, [_hE]: 429 },
    [_m],
    [0]
];
n0_registry.registerError(exports.TooManyRequestsException$, errors_1.TooManyRequestsException);
exports.UnauthorizedException$ = [-3, n0, _UE,
    { [_e]: _c, [_hE]: 401 },
    [_m],
    [0]
];
n0_registry.registerError(exports.UnauthorizedException$, errors_1.UnauthorizedException);
exports.UnexpectedLambdaException$ = [-3, n0, _ULE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.UnexpectedLambdaException$, errors_1.UnexpectedLambdaException);
exports.UnsupportedIdentityProviderException$ = [-3, n0, _UIPE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.UnsupportedIdentityProviderException$, errors_1.UnsupportedIdentityProviderException);
exports.UnsupportedOperationException$ = [-3, n0, _UOE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.UnsupportedOperationException$, errors_1.UnsupportedOperationException);
exports.UnsupportedTokenTypeException$ = [-3, n0, _UTTE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.UnsupportedTokenTypeException$, errors_1.UnsupportedTokenTypeException);
exports.UnsupportedUserStateException$ = [-3, n0, _UUSE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.UnsupportedUserStateException$, errors_1.UnsupportedUserStateException);
exports.UserImportInProgressException$ = [-3, n0, _UIIPE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.UserImportInProgressException$, errors_1.UserImportInProgressException);
exports.UserLambdaValidationException$ = [-3, n0, _ULVE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.UserLambdaValidationException$, errors_1.UserLambdaValidationException);
exports.UsernameExistsException$ = [-3, n0, _UEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.UsernameExistsException$, errors_1.UsernameExistsException);
exports.UserNotConfirmedException$ = [-3, n0, _UNCE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.UserNotConfirmedException$, errors_1.UserNotConfirmedException);
exports.UserNotFoundException$ = [-3, n0, _UNFE,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(exports.UserNotFoundException$, errors_1.UserNotFoundException);
exports.UserPoolAddOnNotEnabledException$ = [-3, n0, _UPAONEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.UserPoolAddOnNotEnabledException$, errors_1.UserPoolAddOnNotEnabledException);
exports.UserPoolTaggingException$ = [-3, n0, _UPTE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.UserPoolTaggingException$, errors_1.UserPoolTaggingException);
exports.WebAuthnChallengeNotFoundException$ = [-3, n0, _WACNFE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.WebAuthnChallengeNotFoundException$, errors_1.WebAuthnChallengeNotFoundException);
exports.WebAuthnClientMismatchException$ = [-3, n0, _WACME,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.WebAuthnClientMismatchException$, errors_1.WebAuthnClientMismatchException);
exports.WebAuthnConfigurationMissingException$ = [-3, n0, _WACMEe,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.WebAuthnConfigurationMissingException$, errors_1.WebAuthnConfigurationMissingException);
exports.WebAuthnCredentialNotSupportedException$ = [-3, n0, _WACNSE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.WebAuthnCredentialNotSupportedException$, errors_1.WebAuthnCredentialNotSupportedException);
exports.WebAuthnNotEnabledException$ = [-3, n0, _WANEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.WebAuthnNotEnabledException$, errors_1.WebAuthnNotEnabledException);
exports.WebAuthnOriginNotAllowedException$ = [-3, n0, _WAONAE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.WebAuthnOriginNotAllowedException$, errors_1.WebAuthnOriginNotAllowedException);
exports.WebAuthnRelyingPartyMismatchException$ = [-3, n0, _WARPME,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.WebAuthnRelyingPartyMismatchException$, errors_1.WebAuthnRelyingPartyMismatchException);
exports.errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var AttributeValueType = [0, n0, _AVT, 8, 0];
var ClientIdType = [0, n0, _CIT, 8, 0];
var ClientSecretType = [0, n0, _CST, 8, 0];
var PasswordType = [0, n0, _PT, 8, 0];
var SecretCodeType = [0, n0, _SCT, 8, 0];
var SecretHashType = [0, n0, _SHT, 8, 0];
var SessionType = [0, n0, _ST, 8, 0];
var SoftwareTokenMFAUserCodeType = [0, n0, _STMFAUCT, 8, 0];
var TokenModelType = [0, n0, _TMT, 8, 0];
var UsernameType = [0, n0, _UT, 8, 0];
exports.AccountRecoverySettingType$ = [3, n0, _ARST,
    0,
    [_RM],
    [() => RecoveryMechanismsType]
];
exports.AccountTakeoverActionsType$ = [3, n0, _ATAT,
    0,
    [_LA, _MA, _HA],
    [() => exports.AccountTakeoverActionType$, () => exports.AccountTakeoverActionType$, () => exports.AccountTakeoverActionType$]
];
exports.AccountTakeoverActionType$ = [3, n0, _ATATc,
    0,
    [_N, _EA],
    [2, 0], 2
];
exports.AccountTakeoverRiskConfigurationType$ = [3, n0, _ATRCT,
    0,
    [_A, _NC],
    [() => exports.AccountTakeoverActionsType$, () => exports.NotifyConfigurationType$], 1
];
exports.AddCustomAttributesRequest$ = [3, n0, _ACAR,
    0,
    [_UPI, _CA],
    [0, () => CustomAttributesListType], 2
];
exports.AddCustomAttributesResponse$ = [3, n0, _ACARd,
    0,
    [],
    []
];
exports.AdminAddUserToGroupRequest$ = [3, n0, _AAUTGR,
    0,
    [_UPI, _U, _GN],
    [0, [() => UsernameType, 0], 0], 3
];
exports.AdminConfirmSignUpRequest$ = [3, n0, _ACSUR,
    0,
    [_UPI, _U, _CM],
    [0, [() => UsernameType, 0], 128 | 0], 2
];
exports.AdminConfirmSignUpResponse$ = [3, n0, _ACSURd,
    0,
    [],
    []
];
exports.AdminCreateUserConfigType$ = [3, n0, _ACUCT,
    0,
    [_AACUO, _UAVD, _IMT],
    [2, 1, () => exports.MessageTemplateType$]
];
exports.AdminCreateUserRequest$ = [3, n0, _ACUR,
    0,
    [_UPI, _U, _UA, _VD, _TP, _FAC, _MAe, _DDM, _CM],
    [0, [() => UsernameType, 0], [() => AttributeListType, 0], [() => AttributeListType, 0], [() => PasswordType, 0], 2, 0, 64 | 0, 128 | 0], 2
];
exports.AdminCreateUserResponse$ = [3, n0, _ACURd,
    0,
    [_Us],
    [[() => exports.UserType$, 0]]
];
exports.AdminDeleteUserAttributesRequest$ = [3, n0, _ADUAR,
    0,
    [_UPI, _U, _UAN],
    [0, [() => UsernameType, 0], 64 | 0], 3
];
exports.AdminDeleteUserAttributesResponse$ = [3, n0, _ADUARd,
    0,
    [],
    []
];
exports.AdminDeleteUserRequest$ = [3, n0, _ADUR,
    0,
    [_UPI, _U],
    [0, [() => UsernameType, 0]], 2
];
exports.AdminDisableProviderForUserRequest$ = [3, n0, _ADPFUR,
    0,
    [_UPI, _Us],
    [0, () => exports.ProviderUserIdentifierType$], 2
];
exports.AdminDisableProviderForUserResponse$ = [3, n0, _ADPFURd,
    0,
    [],
    []
];
exports.AdminDisableUserRequest$ = [3, n0, _ADURd,
    0,
    [_UPI, _U],
    [0, [() => UsernameType, 0]], 2
];
exports.AdminDisableUserResponse$ = [3, n0, _ADURdm,
    0,
    [],
    []
];
exports.AdminEnableUserRequest$ = [3, n0, _AEUR,
    0,
    [_UPI, _U],
    [0, [() => UsernameType, 0]], 2
];
exports.AdminEnableUserResponse$ = [3, n0, _AEURd,
    0,
    [],
    []
];
exports.AdminForgetDeviceRequest$ = [3, n0, _AFDR,
    0,
    [_UPI, _U, _DK],
    [0, [() => UsernameType, 0], 0], 3
];
exports.AdminGetDeviceRequest$ = [3, n0, _AGDR,
    0,
    [_DK, _UPI, _U],
    [0, 0, [() => UsernameType, 0]], 3
];
exports.AdminGetDeviceResponse$ = [3, n0, _AGDRd,
    0,
    [_D],
    [[() => exports.DeviceType$, 0]], 1
];
exports.AdminGetUserRequest$ = [3, n0, _AGUR,
    0,
    [_UPI, _U],
    [0, [() => UsernameType, 0]], 2
];
exports.AdminGetUserResponse$ = [3, n0, _AGURd,
    0,
    [_U, _UA, _UCD, _ULMD, _E, _US, _MFAO, _PMS, _UMFASL],
    [[() => UsernameType, 0], [() => AttributeListType, 0], 4, 4, 2, 0, () => MFAOptionListType, 0, 64 | 0], 1
];
exports.AdminInitiateAuthRequest$ = [3, n0, _AIAR,
    0,
    [_UPI, _CI, _AF, _AP, _CM, _AM, _CD, _S],
    [0, [() => ClientIdType, 0], 0, [() => AuthParametersType, 0], 128 | 0, () => exports.AnalyticsMetadataType$, () => exports.ContextDataType$, [() => SessionType, 0]], 3
];
exports.AdminInitiateAuthResponse$ = [3, n0, _AIARd,
    0,
    [_CN, _S, _CP, _AR, _AC],
    [0, [() => SessionType, 0], 128 | 0, [() => exports.AuthenticationResultType$, 0], 64 | 0]
];
exports.AdminLinkProviderForUserRequest$ = [3, n0, _ALPFUR,
    0,
    [_UPI, _DU, _SU],
    [0, () => exports.ProviderUserIdentifierType$, () => exports.ProviderUserIdentifierType$], 3
];
exports.AdminLinkProviderForUserResponse$ = [3, n0, _ALPFURd,
    0,
    [],
    []
];
exports.AdminListDevicesRequest$ = [3, n0, _ALDR,
    0,
    [_UPI, _U, _L, _PTa],
    [0, [() => UsernameType, 0], 1, 0], 2
];
exports.AdminListDevicesResponse$ = [3, n0, _ALDRd,
    0,
    [_De, _PTa],
    [[() => DeviceListType, 0], 0]
];
exports.AdminListGroupsForUserRequest$ = [3, n0, _ALGFUR,
    0,
    [_U, _UPI, _L, _NT],
    [[() => UsernameType, 0], 0, 1, 0], 2
];
exports.AdminListGroupsForUserResponse$ = [3, n0, _ALGFURd,
    0,
    [_G, _NT],
    [() => GroupListType, 0]
];
exports.AdminListUserAuthEventsRequest$ = [3, n0, _ALUAER,
    0,
    [_UPI, _U, _MR, _NT],
    [0, [() => UsernameType, 0], 1, 0], 2
];
exports.AdminListUserAuthEventsResponse$ = [3, n0, _ALUAERd,
    0,
    [_AE, _NT],
    [() => AuthEventsType, 0]
];
exports.AdminRemoveUserFromGroupRequest$ = [3, n0, _ARUFGR,
    0,
    [_UPI, _U, _GN],
    [0, [() => UsernameType, 0], 0], 3
];
exports.AdminResetUserPasswordRequest$ = [3, n0, _ARUPR,
    0,
    [_UPI, _U, _CM],
    [0, [() => UsernameType, 0], 128 | 0], 2
];
exports.AdminResetUserPasswordResponse$ = [3, n0, _ARUPRd,
    0,
    [],
    []
];
exports.AdminRespondToAuthChallengeRequest$ = [3, n0, _ARTACR,
    0,
    [_UPI, _CI, _CN, _CR, _S, _AM, _CD, _CM],
    [0, [() => ClientIdType, 0], 0, [() => ChallengeResponsesType, 0], [() => SessionType, 0], () => exports.AnalyticsMetadataType$, () => exports.ContextDataType$, 128 | 0], 3
];
exports.AdminRespondToAuthChallengeResponse$ = [3, n0, _ARTACRd,
    0,
    [_CN, _S, _CP, _AR],
    [0, [() => SessionType, 0], 128 | 0, [() => exports.AuthenticationResultType$, 0]]
];
exports.AdminSetUserMFAPreferenceRequest$ = [3, n0, _ASUMFAPR,
    0,
    [_U, _UPI, _SMSMS, _STMS, _EMS],
    [[() => UsernameType, 0], 0, () => exports.SMSMfaSettingsType$, () => exports.SoftwareTokenMfaSettingsType$, () => exports.EmailMfaSettingsType$], 2
];
exports.AdminSetUserMFAPreferenceResponse$ = [3, n0, _ASUMFAPRd,
    0,
    [],
    []
];
exports.AdminSetUserPasswordRequest$ = [3, n0, _ASUPR,
    0,
    [_UPI, _U, _P, _Pe],
    [0, [() => UsernameType, 0], [() => PasswordType, 0], 2], 3
];
exports.AdminSetUserPasswordResponse$ = [3, n0, _ASUPRd,
    0,
    [],
    []
];
exports.AdminSetUserSettingsRequest$ = [3, n0, _ASUSR,
    0,
    [_UPI, _U, _MFAO],
    [0, [() => UsernameType, 0], () => MFAOptionListType], 3
];
exports.AdminSetUserSettingsResponse$ = [3, n0, _ASUSRd,
    0,
    [],
    []
];
exports.AdminUpdateAuthEventFeedbackRequest$ = [3, n0, _AUAEFR,
    0,
    [_UPI, _U, _EI, _FV],
    [0, [() => UsernameType, 0], 0, 0], 4
];
exports.AdminUpdateAuthEventFeedbackResponse$ = [3, n0, _AUAEFRd,
    0,
    [],
    []
];
exports.AdminUpdateDeviceStatusRequest$ = [3, n0, _AUDSR,
    0,
    [_UPI, _U, _DK, _DRS],
    [0, [() => UsernameType, 0], 0, 0], 3
];
exports.AdminUpdateDeviceStatusResponse$ = [3, n0, _AUDSRd,
    0,
    [],
    []
];
exports.AdminUpdateUserAttributesRequest$ = [3, n0, _AUUAR,
    0,
    [_UPI, _U, _UA, _CM],
    [0, [() => UsernameType, 0], [() => AttributeListType, 0], 128 | 0], 3
];
exports.AdminUpdateUserAttributesResponse$ = [3, n0, _AUUARd,
    0,
    [],
    []
];
exports.AdminUserGlobalSignOutRequest$ = [3, n0, _AUGSOR,
    0,
    [_UPI, _U],
    [0, [() => UsernameType, 0]], 2
];
exports.AdminUserGlobalSignOutResponse$ = [3, n0, _AUGSORd,
    0,
    [],
    []
];
exports.AdvancedSecurityAdditionalFlowsType$ = [3, n0, _ASAFT,
    0,
    [_CAM],
    [0]
];
exports.AnalyticsConfigurationType$ = [3, n0, _ACT,
    0,
    [_AI, _AA, _RA, _EIx, _UDS],
    [0, 0, 0, 0, 2]
];
exports.AnalyticsMetadataType$ = [3, n0, _AMT,
    0,
    [_AEI],
    [0]
];
exports.AssetType$ = [3, n0, _AT,
    0,
    [_C, _CMo, _Ex, _B, _RI],
    [0, 0, 0, 21, 0], 3
];
exports.AssociateSoftwareTokenRequest$ = [3, n0, _ASTR,
    0,
    [_ATc, _S],
    [[() => TokenModelType, 0], [() => SessionType, 0]]
];
exports.AssociateSoftwareTokenResponse$ = [3, n0, _ASTRs,
    0,
    [_SC, _S],
    [[() => SecretCodeType, 0], [() => SessionType, 0]]
];
exports.AttributeType$ = [3, n0, _ATt,
    0,
    [_Na, _V],
    [0, [() => AttributeValueType, 0]], 1
];
exports.AuthenticationResultType$ = [3, n0, _ART,
    0,
    [_ATc, _EIxp, _TT, _RT, _IT, _NDM],
    [[() => TokenModelType, 0], 1, 0, [() => TokenModelType, 0], [() => TokenModelType, 0], () => exports.NewDeviceMetadataType$]
];
exports.AuthEventType$ = [3, n0, _AET,
    0,
    [_EI, _ET, _CDr, _ER, _ERv, _CR, _ECD, _EF],
    [0, 0, 4, 0, () => exports.EventRiskType$, () => ChallengeResponseListType, () => exports.EventContextDataType$, () => exports.EventFeedbackType$]
];
exports.ChallengeResponseType$ = [3, n0, _CRT,
    0,
    [_CN, _CRh],
    [0, 0]
];
exports.ChangePasswordRequest$ = [3, n0, _CPR,
    0,
    [_PP, _ATc, _PPr],
    [[() => PasswordType, 0], [() => TokenModelType, 0], [() => PasswordType, 0]], 2
];
exports.ChangePasswordResponse$ = [3, n0, _CPRh,
    0,
    [],
    []
];
exports.CloudWatchLogsConfigurationType$ = [3, n0, _CWLCT,
    0,
    [_LGA],
    [0]
];
exports.CodeDeliveryDetailsType$ = [3, n0, _CDDT,
    0,
    [_Des, _DM, _AN],
    [0, 0, 0]
];
exports.CompleteWebAuthnRegistrationRequest$ = [3, n0, _CWARR,
    0,
    [_ATc, _Cr],
    [[() => TokenModelType, 0], 15], 2
];
exports.CompleteWebAuthnRegistrationResponse$ = [3, n0, _CWARRo,
    0,
    [],
    []
];
exports.CompromisedCredentialsActionsType$ = [3, n0, _CCAT,
    0,
    [_EA],
    [0], 1
];
exports.CompromisedCredentialsRiskConfigurationType$ = [3, n0, _CCRCT,
    0,
    [_A, _EFv],
    [() => exports.CompromisedCredentialsActionsType$, 64 | 0], 1
];
exports.ConfirmDeviceRequest$ = [3, n0, _CDR,
    0,
    [_ATc, _DK, _DSVC, _DN],
    [[() => TokenModelType, 0], 0, () => exports.DeviceSecretVerifierConfigType$, 0], 2
];
exports.ConfirmDeviceResponse$ = [3, n0, _CDRo,
    0,
    [_UCN],
    [2]
];
exports.ConfirmForgotPasswordRequest$ = [3, n0, _CFPR,
    0,
    [_CI, _U, _CC, _P, _SH, _AM, _UCDs, _CM],
    [[() => ClientIdType, 0], [() => UsernameType, 0], 0, [() => PasswordType, 0], [() => SecretHashType, 0], () => exports.AnalyticsMetadataType$, [() => exports.UserContextDataType$, 0], 128 | 0], 4
];
exports.ConfirmForgotPasswordResponse$ = [3, n0, _CFPRo,
    0,
    [],
    []
];
exports.ConfirmSignUpRequest$ = [3, n0, _CSUR,
    0,
    [_CI, _U, _CC, _SH, _FAC, _AM, _UCDs, _CM, _S],
    [[() => ClientIdType, 0], [() => UsernameType, 0], 0, [() => SecretHashType, 0], 2, () => exports.AnalyticsMetadataType$, [() => exports.UserContextDataType$, 0], 128 | 0, [() => SessionType, 0]], 3
];
exports.ConfirmSignUpResponse$ = [3, n0, _CSURo,
    0,
    [_S],
    [[() => SessionType, 0]]
];
exports.ContextDataType$ = [3, n0, _CDT,
    0,
    [_IA, _SN, _SP, _HH, _ED],
    [0, 0, 0, () => HttpHeaderList, 0], 4
];
exports.CreateGroupRequest$ = [3, n0, _CGR,
    0,
    [_GN, _UPI, _Desc, _RA, _Pr],
    [0, 0, 0, 0, 1], 2
];
exports.CreateGroupResponse$ = [3, n0, _CGRr,
    0,
    [_Gr],
    [() => exports.GroupType$]
];
exports.CreateIdentityProviderRequest$ = [3, n0, _CIPR,
    0,
    [_UPI, _PN, _PTr, _PD, _AMt, _II],
    [0, 0, 0, 128 | 0, 128 | 0, 64 | 0], 4
];
exports.CreateIdentityProviderResponse$ = [3, n0, _CIPRr,
    0,
    [_IP],
    [() => exports.IdentityProviderType$], 1
];
exports.CreateManagedLoginBrandingRequest$ = [3, n0, _CMLBR,
    0,
    [_UPI, _CI, _UCPV, _Se, _As],
    [0, [() => ClientIdType, 0], 2, 15, () => AssetListType], 2
];
exports.CreateManagedLoginBrandingResponse$ = [3, n0, _CMLBRr,
    0,
    [_MLB],
    [() => exports.ManagedLoginBrandingType$]
];
exports.CreateResourceServerRequest$ = [3, n0, _CRSR,
    0,
    [_UPI, _I, _Na, _Sc],
    [0, 0, 0, () => ResourceServerScopeListType], 3
];
exports.CreateResourceServerResponse$ = [3, n0, _CRSRr,
    0,
    [_RS],
    [() => exports.ResourceServerType$], 1
];
exports.CreateTermsRequest$ = [3, n0, _CTR,
    0,
    [_UPI, _CI, _TN, _TS, _En, _Li],
    [0, [() => ClientIdType, 0], 0, 0, 0, 128 | 0], 5
];
exports.CreateTermsResponse$ = [3, n0, _CTRr,
    0,
    [_T],
    [[() => exports.TermsType$, 0]]
];
exports.CreateUserImportJobRequest$ = [3, n0, _CUIJR,
    0,
    [_JN, _UPI, _CWLRA],
    [0, 0, 0], 3
];
exports.CreateUserImportJobResponse$ = [3, n0, _CUIJRr,
    0,
    [_UIJ],
    [() => exports.UserImportJobType$]
];
exports.CreateUserPoolClientRequest$ = [3, n0, _CUPCR,
    0,
    [_UPI, _CNl, _GS, _RTV, _ATV, _ITV, _TVU, _RAe, _WA, _EAF, _SIP, _CURL, _LURL, _DRURI, _AOAF, _AOAS, _AOAFUPC, _ACn, _PUEE, _ETR, _EPAUCD, _ASV, _RTR],
    [0, 0, 2, 1, 1, 1, () => exports.TokenValidityUnitsType$, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 0, 64 | 0, 64 | 0, 2, () => exports.AnalyticsConfigurationType$, 0, 2, 2, 1, () => exports.RefreshTokenRotationType$], 2
];
exports.CreateUserPoolClientResponse$ = [3, n0, _CUPCRr,
    0,
    [_UPC],
    [[() => exports.UserPoolClientType$, 0]]
];
exports.CreateUserPoolDomainRequest$ = [3, n0, _CUPDR,
    0,
    [_Do, _UPI, _MLV, _CDC],
    [0, 0, 1, () => exports.CustomDomainConfigType$], 2
];
exports.CreateUserPoolDomainResponse$ = [3, n0, _CUPDRr,
    0,
    [_MLV, _CFD],
    [1, 0]
];
exports.CreateUserPoolRequest$ = [3, n0, _CUPR,
    0,
    [_PNo, _Po, _DP, _LC, _AVA, _AAl, _UAs, _SVM, _EVM, _EVS, _VMT, _SAM, _MC, _UAUS, _DC, _EC, _SCm, _UPT, _ACUC, _Sch, _UPAO, _UC, _ARS, _UPTs],
    [0, () => exports.UserPoolPolicyType$, 0, () => exports.LambdaConfigType$, 64 | 0, 64 | 0, 64 | 0, 0, 0, 0, () => exports.VerificationMessageTemplateType$, 0, 0, () => exports.UserAttributeUpdateSettingsType$, () => exports.DeviceConfigurationType$, () => exports.EmailConfigurationType$, () => exports.SmsConfigurationType$, 128 | 0, () => exports.AdminCreateUserConfigType$, () => SchemaAttributesListType, () => exports.UserPoolAddOnsType$, () => exports.UsernameConfigurationType$, () => exports.AccountRecoverySettingType$, 0], 1
];
exports.CreateUserPoolResponse$ = [3, n0, _CUPRr,
    0,
    [_UP],
    [() => exports.UserPoolType$]
];
exports.CustomDomainConfigType$ = [3, n0, _CDCT,
    0,
    [_CAe],
    [0], 1
];
exports.CustomEmailLambdaVersionConfigType$ = [3, n0, _CELVCT,
    0,
    [_LV, _LAa],
    [0, 0], 2
];
exports.CustomSMSLambdaVersionConfigType$ = [3, n0, _CSMSLVCT,
    0,
    [_LV, _LAa],
    [0, 0], 2
];
exports.DeleteGroupRequest$ = [3, n0, _DGR,
    0,
    [_GN, _UPI],
    [0, 0], 2
];
exports.DeleteIdentityProviderRequest$ = [3, n0, _DIPR,
    0,
    [_UPI, _PN],
    [0, 0], 2
];
exports.DeleteManagedLoginBrandingRequest$ = [3, n0, _DMLBR,
    0,
    [_MLBI, _UPI],
    [0, 0], 2
];
exports.DeleteResourceServerRequest$ = [3, n0, _DRSR,
    0,
    [_UPI, _I],
    [0, 0], 2
];
exports.DeleteTermsRequest$ = [3, n0, _DTR,
    0,
    [_TI, _UPI],
    [0, 0], 2
];
exports.DeleteUserAttributesRequest$ = [3, n0, _DUAR,
    0,
    [_UAN, _ATc],
    [64 | 0, [() => TokenModelType, 0]], 2
];
exports.DeleteUserAttributesResponse$ = [3, n0, _DUARe,
    0,
    [],
    []
];
exports.DeleteUserPoolClientRequest$ = [3, n0, _DUPCR,
    0,
    [_UPI, _CI],
    [0, [() => ClientIdType, 0]], 2
];
exports.DeleteUserPoolDomainRequest$ = [3, n0, _DUPDR,
    0,
    [_Do, _UPI],
    [0, 0], 2
];
exports.DeleteUserPoolDomainResponse$ = [3, n0, _DUPDRe,
    0,
    [],
    []
];
exports.DeleteUserPoolRequest$ = [3, n0, _DUPR,
    0,
    [_UPI],
    [0], 1
];
exports.DeleteUserRequest$ = [3, n0, _DUR,
    0,
    [_ATc],
    [[() => TokenModelType, 0]], 1
];
exports.DeleteWebAuthnCredentialRequest$ = [3, n0, _DWACR,
    0,
    [_ATc, _CIr],
    [[() => TokenModelType, 0], 0], 2
];
exports.DeleteWebAuthnCredentialResponse$ = [3, n0, _DWACRe,
    0,
    [],
    []
];
exports.DescribeIdentityProviderRequest$ = [3, n0, _DIPRe,
    0,
    [_UPI, _PN],
    [0, 0], 2
];
exports.DescribeIdentityProviderResponse$ = [3, n0, _DIPRes,
    0,
    [_IP],
    [() => exports.IdentityProviderType$], 1
];
exports.DescribeManagedLoginBrandingByClientRequest$ = [3, n0, _DMLBBCR,
    0,
    [_UPI, _CI, _RMR],
    [0, [() => ClientIdType, 0], 2], 2
];
exports.DescribeManagedLoginBrandingByClientResponse$ = [3, n0, _DMLBBCRe,
    0,
    [_MLB],
    [() => exports.ManagedLoginBrandingType$]
];
exports.DescribeManagedLoginBrandingRequest$ = [3, n0, _DMLBRe,
    0,
    [_UPI, _MLBI, _RMR],
    [0, 0, 2], 2
];
exports.DescribeManagedLoginBrandingResponse$ = [3, n0, _DMLBRes,
    0,
    [_MLB],
    [() => exports.ManagedLoginBrandingType$]
];
exports.DescribeResourceServerRequest$ = [3, n0, _DRSRe,
    0,
    [_UPI, _I],
    [0, 0], 2
];
exports.DescribeResourceServerResponse$ = [3, n0, _DRSRes,
    0,
    [_RS],
    [() => exports.ResourceServerType$], 1
];
exports.DescribeRiskConfigurationRequest$ = [3, n0, _DRCR,
    0,
    [_UPI, _CI],
    [0, [() => ClientIdType, 0]], 1
];
exports.DescribeRiskConfigurationResponse$ = [3, n0, _DRCRe,
    0,
    [_RC],
    [[() => exports.RiskConfigurationType$, 0]], 1
];
exports.DescribeTermsRequest$ = [3, n0, _DTRe,
    0,
    [_TI, _UPI],
    [0, 0], 2
];
exports.DescribeTermsResponse$ = [3, n0, _DTRes,
    0,
    [_T],
    [[() => exports.TermsType$, 0]]
];
exports.DescribeUserImportJobRequest$ = [3, n0, _DUIJR,
    0,
    [_UPI, _JI],
    [0, 0], 2
];
exports.DescribeUserImportJobResponse$ = [3, n0, _DUIJRe,
    0,
    [_UIJ],
    [() => exports.UserImportJobType$]
];
exports.DescribeUserPoolClientRequest$ = [3, n0, _DUPCRe,
    0,
    [_UPI, _CI],
    [0, [() => ClientIdType, 0]], 2
];
exports.DescribeUserPoolClientResponse$ = [3, n0, _DUPCRes,
    0,
    [_UPC],
    [[() => exports.UserPoolClientType$, 0]]
];
exports.DescribeUserPoolDomainRequest$ = [3, n0, _DUPDRes,
    0,
    [_Do],
    [0], 1
];
exports.DescribeUserPoolDomainResponse$ = [3, n0, _DUPDResc,
    0,
    [_DD],
    [() => exports.DomainDescriptionType$]
];
exports.DescribeUserPoolRequest$ = [3, n0, _DUPRe,
    0,
    [_UPI],
    [0], 1
];
exports.DescribeUserPoolResponse$ = [3, n0, _DUPRes,
    0,
    [_UP],
    [() => exports.UserPoolType$]
];
exports.DeviceConfigurationType$ = [3, n0, _DCT,
    0,
    [_CROND, _DOROUP],
    [2, 2]
];
exports.DeviceSecretVerifierConfigType$ = [3, n0, _DSVCT,
    0,
    [_PV, _Sa],
    [0, 0]
];
exports.DeviceType$ = [3, n0, _DT,
    0,
    [_DK, _DA, _DCD, _DLMD, _DLAD],
    [0, [() => AttributeListType, 0], 4, 4, 4]
];
exports.DomainDescriptionType$ = [3, n0, _DDT,
    0,
    [_UPI, _AWSAI, _Do, _SB, _CFDl, _Ve, _St, _CDC, _MLV],
    [0, 0, 0, 0, 0, 0, 0, () => exports.CustomDomainConfigType$, 1]
];
exports.EmailConfigurationType$ = [3, n0, _ECT,
    0,
    [_SA, _RTEA, _ESA, _F, _CS],
    [0, 0, 0, 0, 0]
];
exports.EmailMfaConfigType$ = [3, n0, _EMCT,
    0,
    [_M, _Su],
    [0, 0]
];
exports.EmailMfaSettingsType$ = [3, n0, _EMST,
    0,
    [_E, _PM],
    [2, 2]
];
exports.EventContextDataType$ = [3, n0, _ECDT,
    0,
    [_IA, _DN, _Ti, _Ci, _Co],
    [0, 0, 0, 0, 0]
];
exports.EventFeedbackType$ = [3, n0, _EFT,
    0,
    [_FV, _Pro, _FD],
    [0, 0, 4], 2
];
exports.EventRiskType$ = [3, n0, _ERT,
    0,
    [_RD, _RL, _CCD],
    [0, 0, 2]
];
exports.FirehoseConfigurationType$ = [3, n0, _FCT,
    0,
    [_SAt],
    [0]
];
exports.ForgetDeviceRequest$ = [3, n0, _FDR,
    0,
    [_DK, _ATc],
    [0, [() => TokenModelType, 0]], 1
];
exports.ForgotPasswordRequest$ = [3, n0, _FPR,
    0,
    [_CI, _U, _SH, _UCDs, _AM, _CM],
    [[() => ClientIdType, 0], [() => UsernameType, 0], [() => SecretHashType, 0], [() => exports.UserContextDataType$, 0], () => exports.AnalyticsMetadataType$, 128 | 0], 2
];
exports.ForgotPasswordResponse$ = [3, n0, _FPRo,
    0,
    [_CDD],
    [() => exports.CodeDeliveryDetailsType$]
];
exports.GetCSVHeaderRequest$ = [3, n0, _GCSVHR,
    0,
    [_UPI],
    [0], 1
];
exports.GetCSVHeaderResponse$ = [3, n0, _GCSVHRe,
    0,
    [_UPI, _CSVH],
    [0, 64 | 0]
];
exports.GetDeviceRequest$ = [3, n0, _GDR,
    0,
    [_DK, _ATc],
    [0, [() => TokenModelType, 0]], 1
];
exports.GetDeviceResponse$ = [3, n0, _GDRe,
    0,
    [_D],
    [[() => exports.DeviceType$, 0]], 1
];
exports.GetGroupRequest$ = [3, n0, _GGR,
    0,
    [_GN, _UPI],
    [0, 0], 2
];
exports.GetGroupResponse$ = [3, n0, _GGRe,
    0,
    [_Gr],
    [() => exports.GroupType$]
];
exports.GetIdentityProviderByIdentifierRequest$ = [3, n0, _GIPBIR,
    0,
    [_UPI, _IId],
    [0, 0], 2
];
exports.GetIdentityProviderByIdentifierResponse$ = [3, n0, _GIPBIRe,
    0,
    [_IP],
    [() => exports.IdentityProviderType$], 1
];
exports.GetLogDeliveryConfigurationRequest$ = [3, n0, _GLDCR,
    0,
    [_UPI],
    [0], 1
];
exports.GetLogDeliveryConfigurationResponse$ = [3, n0, _GLDCRe,
    0,
    [_LDC],
    [() => exports.LogDeliveryConfigurationType$]
];
exports.GetSigningCertificateRequest$ = [3, n0, _GSCR,
    0,
    [_UPI],
    [0], 1
];
exports.GetSigningCertificateResponse$ = [3, n0, _GSCRe,
    0,
    [_Ce],
    [0]
];
exports.GetTokensFromRefreshTokenRequest$ = [3, n0, _GTFRTR,
    0,
    [_RT, _CI, _CSl, _DK, _CM],
    [[() => TokenModelType, 0], [() => ClientIdType, 0], [() => ClientSecretType, 0], 0, 128 | 0], 2
];
exports.GetTokensFromRefreshTokenResponse$ = [3, n0, _GTFRTRe,
    0,
    [_AR],
    [[() => exports.AuthenticationResultType$, 0]]
];
exports.GetUICustomizationRequest$ = [3, n0, _GUICR,
    0,
    [_UPI, _CI],
    [0, [() => ClientIdType, 0]], 1
];
exports.GetUICustomizationResponse$ = [3, n0, _GUICRe,
    0,
    [_UIC],
    [[() => exports.UICustomizationType$, 0]], 1
];
exports.GetUserAttributeVerificationCodeRequest$ = [3, n0, _GUAVCR,
    0,
    [_ATc, _AN, _CM],
    [[() => TokenModelType, 0], 0, 128 | 0], 2
];
exports.GetUserAttributeVerificationCodeResponse$ = [3, n0, _GUAVCRe,
    0,
    [_CDD],
    [() => exports.CodeDeliveryDetailsType$]
];
exports.GetUserAuthFactorsRequest$ = [3, n0, _GUAFR,
    0,
    [_ATc],
    [[() => TokenModelType, 0]], 1
];
exports.GetUserAuthFactorsResponse$ = [3, n0, _GUAFRe,
    0,
    [_U, _PMS, _UMFASL, _CUAF],
    [[() => UsernameType, 0], 0, 64 | 0, 64 | 0], 1
];
exports.GetUserPoolMfaConfigRequest$ = [3, n0, _GUPMCR,
    0,
    [_UPI],
    [0], 1
];
exports.GetUserPoolMfaConfigResponse$ = [3, n0, _GUPMCRe,
    0,
    [_SMC, _STMC, _EMC, _MC, _WAC],
    [() => exports.SmsMfaConfigType$, () => exports.SoftwareTokenMfaConfigType$, () => exports.EmailMfaConfigType$, 0, () => exports.WebAuthnConfigurationType$]
];
exports.GetUserRequest$ = [3, n0, _GUR,
    0,
    [_ATc],
    [[() => TokenModelType, 0]], 1
];
exports.GetUserResponse$ = [3, n0, _GURe,
    0,
    [_U, _UA, _MFAO, _PMS, _UMFASL],
    [[() => UsernameType, 0], [() => AttributeListType, 0], () => MFAOptionListType, 0, 64 | 0], 2
];
exports.GlobalSignOutRequest$ = [3, n0, _GSOR,
    0,
    [_ATc],
    [[() => TokenModelType, 0]], 1
];
exports.GlobalSignOutResponse$ = [3, n0, _GSORl,
    0,
    [],
    []
];
exports.GroupType$ = [3, n0, _GT,
    0,
    [_GN, _UPI, _Desc, _RA, _Pr, _LMD, _CDr],
    [0, 0, 0, 0, 1, 4, 4]
];
exports.HttpHeader$ = [3, n0, _HHt,
    0,
    [_hN, _hV],
    [0, 0]
];
exports.IdentityProviderType$ = [3, n0, _IPT,
    0,
    [_UPI, _PN, _PTr, _PD, _AMt, _II, _LMD, _CDr],
    [0, 0, 0, 128 | 0, 128 | 0, 64 | 0, 4, 4]
];
exports.InboundFederationLambdaType$ = [3, n0, _IFLT,
    0,
    [_LV, _LAa],
    [0, 0], 2
];
exports.InitiateAuthRequest$ = [3, n0, _IAR,
    0,
    [_AF, _CI, _AP, _CM, _AM, _UCDs, _S],
    [0, [() => ClientIdType, 0], [() => AuthParametersType, 0], 128 | 0, () => exports.AnalyticsMetadataType$, [() => exports.UserContextDataType$, 0], [() => SessionType, 0]], 2
];
exports.InitiateAuthResponse$ = [3, n0, _IARn,
    0,
    [_CN, _S, _CP, _AR, _AC],
    [0, [() => SessionType, 0], 128 | 0, [() => exports.AuthenticationResultType$, 0], 64 | 0]
];
exports.LambdaConfigType$ = [3, n0, _LCT,
    0,
    [_PSU, _CMu, _PC, _PA, _PAo, _DAC, _CAC, _VACR, _PTG, _UM, _PTGC, _CSMSS, _CES, _KMSKID, _IF],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, () => exports.PreTokenGenerationVersionConfigType$, () => exports.CustomSMSLambdaVersionConfigType$, () => exports.CustomEmailLambdaVersionConfigType$, 0, () => exports.InboundFederationLambdaType$]
];
exports.ListDevicesRequest$ = [3, n0, _LDR,
    0,
    [_ATc, _L, _PTa],
    [[() => TokenModelType, 0], 1, 0], 1
];
exports.ListDevicesResponse$ = [3, n0, _LDRi,
    0,
    [_De, _PTa],
    [[() => DeviceListType, 0], 0]
];
exports.ListGroupsRequest$ = [3, n0, _LGR,
    0,
    [_UPI, _L, _NT],
    [0, 1, 0], 1
];
exports.ListGroupsResponse$ = [3, n0, _LGRi,
    0,
    [_G, _NT],
    [() => GroupListType, 0]
];
exports.ListIdentityProvidersRequest$ = [3, n0, _LIPR,
    0,
    [_UPI, _MR, _NT],
    [0, 1, 0], 1
];
exports.ListIdentityProvidersResponse$ = [3, n0, _LIPRi,
    0,
    [_Prov, _NT],
    [() => ProvidersListType, 0], 1
];
exports.ListResourceServersRequest$ = [3, n0, _LRSR,
    0,
    [_UPI, _MR, _NT],
    [0, 1, 0], 1
];
exports.ListResourceServersResponse$ = [3, n0, _LRSRi,
    0,
    [_RSe, _NT],
    [() => ResourceServersListType, 0], 1
];
exports.ListTagsForResourceRequest$ = [3, n0, _LTFRR,
    0,
    [_RAes],
    [0], 1
];
exports.ListTagsForResourceResponse$ = [3, n0, _LTFRRi,
    0,
    [_Ta],
    [128 | 0]
];
exports.ListTermsRequest$ = [3, n0, _LTR,
    0,
    [_UPI, _MR, _NT],
    [0, 1, 0], 1
];
exports.ListTermsResponse$ = [3, n0, _LTRi,
    0,
    [_T, _NT],
    [() => TermsDescriptionListType, 0], 1
];
exports.ListUserImportJobsRequest$ = [3, n0, _LUIJR,
    0,
    [_UPI, _MR, _PTa],
    [0, 1, 0], 2
];
exports.ListUserImportJobsResponse$ = [3, n0, _LUIJRi,
    0,
    [_UIJs, _PTa],
    [() => UserImportJobsListType, 0]
];
exports.ListUserPoolClientsRequest$ = [3, n0, _LUPCR,
    0,
    [_UPI, _MR, _NT],
    [0, 1, 0], 1
];
exports.ListUserPoolClientsResponse$ = [3, n0, _LUPCRi,
    0,
    [_UPCs, _NT],
    [[() => UserPoolClientListType, 0], 0]
];
exports.ListUserPoolsRequest$ = [3, n0, _LUPR,
    0,
    [_MR, _NT],
    [1, 0], 1
];
exports.ListUserPoolsResponse$ = [3, n0, _LUPRi,
    0,
    [_UPs, _NT],
    [() => UserPoolListType, 0]
];
exports.ListUsersInGroupRequest$ = [3, n0, _LUIGR,
    0,
    [_UPI, _GN, _L, _NT],
    [0, 0, 1, 0], 2
];
exports.ListUsersInGroupResponse$ = [3, n0, _LUIGRi,
    0,
    [_Use, _NT],
    [[() => UsersListType, 0], 0]
];
exports.ListUsersRequest$ = [3, n0, _LUR,
    0,
    [_UPI, _ATG, _L, _PTa, _Fi],
    [0, 64 | 0, 1, 0, 0], 1
];
exports.ListUsersResponse$ = [3, n0, _LURi,
    0,
    [_Use, _PTa],
    [[() => UsersListType, 0], 0]
];
exports.ListWebAuthnCredentialsRequest$ = [3, n0, _LWACR,
    0,
    [_ATc, _NT, _MR],
    [[() => TokenModelType, 0], 0, 1], 1
];
exports.ListWebAuthnCredentialsResponse$ = [3, n0, _LWACRi,
    0,
    [_Cre, _NT],
    [() => WebAuthnCredentialDescriptionListType, 0], 1
];
exports.LogConfigurationType$ = [3, n0, _LCTo,
    0,
    [_LL, _ES, _CWLC, _SCo, _FC],
    [0, 0, () => exports.CloudWatchLogsConfigurationType$, () => exports.S3ConfigurationType$, () => exports.FirehoseConfigurationType$], 2
];
exports.LogDeliveryConfigurationType$ = [3, n0, _LDCT,
    0,
    [_UPI, _LCo],
    [0, () => LogConfigurationListType], 2
];
exports.ManagedLoginBrandingType$ = [3, n0, _MLBT,
    0,
    [_MLBI, _UPI, _UCPV, _Se, _As, _CDr, _LMD],
    [0, 0, 2, 15, () => AssetListType, 4, 4]
];
exports.MessageTemplateType$ = [3, n0, _MTT,
    0,
    [_SMSM, _EM, _ESm],
    [0, 0, 0]
];
exports.MFAOptionType$ = [3, n0, _MFAOT,
    0,
    [_DM, _AN],
    [0, 0]
];
exports.NewDeviceMetadataType$ = [3, n0, _NDMT,
    0,
    [_DK, _DGK],
    [0, 0]
];
exports.NotifyConfigurationType$ = [3, n0, _NCT,
    0,
    [_SA, _F, _RTe, _BE, _NAEo, _ME],
    [0, 0, 0, () => exports.NotifyEmailType$, () => exports.NotifyEmailType$, () => exports.NotifyEmailType$], 1
];
exports.NotifyEmailType$ = [3, n0, _NET,
    0,
    [_Su, _HB, _TB],
    [0, 0, 0], 1
];
exports.NumberAttributeConstraintsType$ = [3, n0, _NACT,
    0,
    [_MV, _MVa],
    [0, 0]
];
exports.PasswordPolicyType$ = [3, n0, _PPT,
    0,
    [_ML, _RU, _RLe, _RN, _RSeq, _PHS, _TPVD],
    [1, 2, 2, 2, 2, 1, 1]
];
exports.PreTokenGenerationVersionConfigType$ = [3, n0, _PTGVCT,
    0,
    [_LV, _LAa],
    [0, 0], 2
];
exports.ProviderDescription$ = [3, n0, _PDr,
    0,
    [_PN, _PTr, _LMD, _CDr],
    [0, 0, 4, 4]
];
exports.ProviderUserIdentifierType$ = [3, n0, _PUIT,
    0,
    [_PN, _PAN, _PAV],
    [0, 0, 0]
];
exports.RecoveryOptionType$ = [3, n0, _ROT,
    0,
    [_Pri, _Na],
    [1, 0], 2
];
exports.RefreshTokenRotationType$ = [3, n0, _RTRT,
    0,
    [_Fe, _RGPS],
    [0, 1], 1
];
exports.ResendConfirmationCodeRequest$ = [3, n0, _RCCR,
    0,
    [_CI, _U, _SH, _UCDs, _AM, _CM],
    [[() => ClientIdType, 0], [() => UsernameType, 0], [() => SecretHashType, 0], [() => exports.UserContextDataType$, 0], () => exports.AnalyticsMetadataType$, 128 | 0], 2
];
exports.ResendConfirmationCodeResponse$ = [3, n0, _RCCRe,
    0,
    [_CDD],
    [() => exports.CodeDeliveryDetailsType$]
];
exports.ResourceServerScopeType$ = [3, n0, _RSST,
    0,
    [_SNc, _SD],
    [0, 0], 2
];
exports.ResourceServerType$ = [3, n0, _RST,
    0,
    [_UPI, _I, _Na, _Sc],
    [0, 0, 0, () => ResourceServerScopeListType]
];
exports.RespondToAuthChallengeRequest$ = [3, n0, _RTACR,
    0,
    [_CI, _CN, _S, _CR, _AM, _UCDs, _CM],
    [[() => ClientIdType, 0], 0, [() => SessionType, 0], [() => ChallengeResponsesType, 0], () => exports.AnalyticsMetadataType$, [() => exports.UserContextDataType$, 0], 128 | 0], 2
];
exports.RespondToAuthChallengeResponse$ = [3, n0, _RTACRe,
    0,
    [_CN, _S, _CP, _AR],
    [0, [() => SessionType, 0], 128 | 0, [() => exports.AuthenticationResultType$, 0]]
];
exports.RevokeTokenRequest$ = [3, n0, _RTRe,
    0,
    [_To, _CI, _CSl],
    [[() => TokenModelType, 0], [() => ClientIdType, 0], [() => ClientSecretType, 0]], 2
];
exports.RevokeTokenResponse$ = [3, n0, _RTRev,
    0,
    [],
    []
];
exports.RiskConfigurationType$ = [3, n0, _RCT,
    0,
    [_UPI, _CI, _CCRC, _ATRC, _REC, _LMD],
    [0, [() => ClientIdType, 0], () => exports.CompromisedCredentialsRiskConfigurationType$, () => exports.AccountTakeoverRiskConfigurationType$, () => exports.RiskExceptionConfigurationType$, 4]
];
exports.RiskExceptionConfigurationType$ = [3, n0, _RECT,
    0,
    [_BIPRL, _SIPRL],
    [64 | 0, 64 | 0]
];
exports.S3ConfigurationType$ = [3, n0, _SCTo,
    0,
    [_BA],
    [0]
];
exports.SchemaAttributeType$ = [3, n0, _SAT,
    0,
    [_Na, _ADT, _DOA, _Mu, _R, _NAC, _SAC],
    [0, 0, 2, 2, 2, () => exports.NumberAttributeConstraintsType$, () => exports.StringAttributeConstraintsType$]
];
exports.SetLogDeliveryConfigurationRequest$ = [3, n0, _SLDCR,
    0,
    [_UPI, _LCo],
    [0, () => LogConfigurationListType], 2
];
exports.SetLogDeliveryConfigurationResponse$ = [3, n0, _SLDCRe,
    0,
    [_LDC],
    [() => exports.LogDeliveryConfigurationType$]
];
exports.SetRiskConfigurationRequest$ = [3, n0, _SRCR,
    0,
    [_UPI, _CI, _CCRC, _ATRC, _REC],
    [0, [() => ClientIdType, 0], () => exports.CompromisedCredentialsRiskConfigurationType$, () => exports.AccountTakeoverRiskConfigurationType$, () => exports.RiskExceptionConfigurationType$], 1
];
exports.SetRiskConfigurationResponse$ = [3, n0, _SRCRe,
    0,
    [_RC],
    [[() => exports.RiskConfigurationType$, 0]], 1
];
exports.SetUICustomizationRequest$ = [3, n0, _SUICR,
    0,
    [_UPI, _CI, _CSS, _IFm],
    [0, [() => ClientIdType, 0], 0, 21], 1
];
exports.SetUICustomizationResponse$ = [3, n0, _SUICRe,
    0,
    [_UIC],
    [[() => exports.UICustomizationType$, 0]], 1
];
exports.SetUserMFAPreferenceRequest$ = [3, n0, _SUMFAPR,
    0,
    [_ATc, _SMSMS, _STMS, _EMS],
    [[() => TokenModelType, 0], () => exports.SMSMfaSettingsType$, () => exports.SoftwareTokenMfaSettingsType$, () => exports.EmailMfaSettingsType$], 1
];
exports.SetUserMFAPreferenceResponse$ = [3, n0, _SUMFAPRe,
    0,
    [],
    []
];
exports.SetUserPoolMfaConfigRequest$ = [3, n0, _SUPMCR,
    0,
    [_UPI, _SMC, _STMC, _EMC, _MC, _WAC],
    [0, () => exports.SmsMfaConfigType$, () => exports.SoftwareTokenMfaConfigType$, () => exports.EmailMfaConfigType$, 0, () => exports.WebAuthnConfigurationType$], 1
];
exports.SetUserPoolMfaConfigResponse$ = [3, n0, _SUPMCRe,
    0,
    [_SMC, _STMC, _EMC, _MC, _WAC],
    [() => exports.SmsMfaConfigType$, () => exports.SoftwareTokenMfaConfigType$, () => exports.EmailMfaConfigType$, 0, () => exports.WebAuthnConfigurationType$]
];
exports.SetUserSettingsRequest$ = [3, n0, _SUSR,
    0,
    [_ATc, _MFAO],
    [[() => TokenModelType, 0], () => MFAOptionListType], 2
];
exports.SetUserSettingsResponse$ = [3, n0, _SUSRe,
    0,
    [],
    []
];
exports.SignInPolicyType$ = [3, n0, _SIPT,
    0,
    [_AFAF],
    [64 | 0]
];
exports.SignUpRequest$ = [3, n0, _SUR,
    0,
    [_CI, _U, _SH, _P, _UA, _VD, _AM, _UCDs, _CM],
    [[() => ClientIdType, 0], [() => UsernameType, 0], [() => SecretHashType, 0], [() => PasswordType, 0], [() => AttributeListType, 0], [() => AttributeListType, 0], () => exports.AnalyticsMetadataType$, [() => exports.UserContextDataType$, 0], 128 | 0], 2
];
exports.SignUpResponse$ = [3, n0, _SURi,
    0,
    [_UCs, _USs, _CDD, _S],
    [2, 0, () => exports.CodeDeliveryDetailsType$, [() => SessionType, 0]], 2
];
exports.SmsConfigurationType$ = [3, n0, _SCTm,
    0,
    [_SCA, _EIx, _SR],
    [0, 0, 0], 1
];
exports.SmsMfaConfigType$ = [3, n0, _SMCT,
    0,
    [_SAM, _SCm],
    [0, () => exports.SmsConfigurationType$]
];
exports.SMSMfaSettingsType$ = [3, n0, _SMSMST,
    0,
    [_E, _PM],
    [2, 2]
];
exports.SoftwareTokenMfaConfigType$ = [3, n0, _STMCT,
    0,
    [_E],
    [2]
];
exports.SoftwareTokenMfaSettingsType$ = [3, n0, _STMST,
    0,
    [_E, _PM],
    [2, 2]
];
exports.StartUserImportJobRequest$ = [3, n0, _SUIJR,
    0,
    [_UPI, _JI],
    [0, 0], 2
];
exports.StartUserImportJobResponse$ = [3, n0, _SUIJRt,
    0,
    [_UIJ],
    [() => exports.UserImportJobType$]
];
exports.StartWebAuthnRegistrationRequest$ = [3, n0, _SWARR,
    0,
    [_ATc],
    [[() => TokenModelType, 0]], 1
];
exports.StartWebAuthnRegistrationResponse$ = [3, n0, _SWARRt,
    0,
    [_CCO],
    [15], 1
];
exports.StopUserImportJobRequest$ = [3, n0, _SUIJRto,
    0,
    [_UPI, _JI],
    [0, 0], 2
];
exports.StopUserImportJobResponse$ = [3, n0, _SUIJRtop,
    0,
    [_UIJ],
    [() => exports.UserImportJobType$]
];
exports.StringAttributeConstraintsType$ = [3, n0, _SACT,
    0,
    [_MLi, _MLa],
    [0, 0]
];
exports.TagResourceRequest$ = [3, n0, _TRR,
    0,
    [_RAes, _Ta],
    [0, 128 | 0], 2
];
exports.TagResourceResponse$ = [3, n0, _TRRa,
    0,
    [],
    []
];
exports.TermsDescriptionType$ = [3, n0, _TDT,
    0,
    [_TI, _TN, _En, _CDr, _LMD],
    [0, 0, 0, 4, 4], 5
];
exports.TermsType$ = [3, n0, _TTe,
    0,
    [_TI, _UPI, _CI, _TN, _TS, _En, _Li, _CDr, _LMD],
    [0, 0, [() => ClientIdType, 0], 0, 0, 0, 128 | 0, 4, 4], 9
];
exports.TokenValidityUnitsType$ = [3, n0, _TVUT,
    0,
    [_ATc, _IT, _RT],
    [0, 0, 0]
];
exports.UICustomizationType$ = [3, n0, _UICT,
    0,
    [_UPI, _CI, _IU, _CSS, _CSSV, _LMD, _CDr],
    [0, [() => ClientIdType, 0], 0, 0, 0, 4, 4]
];
exports.UntagResourceRequest$ = [3, n0, _URR,
    0,
    [_RAes, _TK],
    [0, 64 | 0], 2
];
exports.UntagResourceResponse$ = [3, n0, _URRn,
    0,
    [],
    []
];
exports.UpdateAuthEventFeedbackRequest$ = [3, n0, _UAEFR,
    0,
    [_UPI, _U, _EI, _FT, _FV],
    [0, [() => UsernameType, 0], 0, [() => TokenModelType, 0], 0], 5
];
exports.UpdateAuthEventFeedbackResponse$ = [3, n0, _UAEFRp,
    0,
    [],
    []
];
exports.UpdateDeviceStatusRequest$ = [3, n0, _UDSR,
    0,
    [_ATc, _DK, _DRS],
    [[() => TokenModelType, 0], 0, 0], 2
];
exports.UpdateDeviceStatusResponse$ = [3, n0, _UDSRp,
    0,
    [],
    []
];
exports.UpdateGroupRequest$ = [3, n0, _UGR,
    0,
    [_GN, _UPI, _Desc, _RA, _Pr],
    [0, 0, 0, 0, 1], 2
];
exports.UpdateGroupResponse$ = [3, n0, _UGRp,
    0,
    [_Gr],
    [() => exports.GroupType$]
];
exports.UpdateIdentityProviderRequest$ = [3, n0, _UIPR,
    0,
    [_UPI, _PN, _PD, _AMt, _II],
    [0, 0, 128 | 0, 128 | 0, 64 | 0], 2
];
exports.UpdateIdentityProviderResponse$ = [3, n0, _UIPRp,
    0,
    [_IP],
    [() => exports.IdentityProviderType$], 1
];
exports.UpdateManagedLoginBrandingRequest$ = [3, n0, _UMLBR,
    0,
    [_UPI, _MLBI, _UCPV, _Se, _As],
    [0, 0, 2, 15, () => AssetListType]
];
exports.UpdateManagedLoginBrandingResponse$ = [3, n0, _UMLBRp,
    0,
    [_MLB],
    [() => exports.ManagedLoginBrandingType$]
];
exports.UpdateResourceServerRequest$ = [3, n0, _URSR,
    0,
    [_UPI, _I, _Na, _Sc],
    [0, 0, 0, () => ResourceServerScopeListType], 3
];
exports.UpdateResourceServerResponse$ = [3, n0, _URSRp,
    0,
    [_RS],
    [() => exports.ResourceServerType$], 1
];
exports.UpdateTermsRequest$ = [3, n0, _UTR,
    0,
    [_TI, _UPI, _TN, _TS, _En, _Li],
    [0, 0, 0, 0, 0, 128 | 0], 2
];
exports.UpdateTermsResponse$ = [3, n0, _UTRp,
    0,
    [_T],
    [[() => exports.TermsType$, 0]]
];
exports.UpdateUserAttributesRequest$ = [3, n0, _UUAR,
    0,
    [_UA, _ATc, _CM],
    [[() => AttributeListType, 0], [() => TokenModelType, 0], 128 | 0], 2
];
exports.UpdateUserAttributesResponse$ = [3, n0, _UUARp,
    0,
    [_CDDL],
    [() => CodeDeliveryDetailsListType]
];
exports.UpdateUserPoolClientRequest$ = [3, n0, _UUPCR,
    0,
    [_UPI, _CI, _CNl, _RTV, _ATV, _ITV, _TVU, _RAe, _WA, _EAF, _SIP, _CURL, _LURL, _DRURI, _AOAF, _AOAS, _AOAFUPC, _ACn, _PUEE, _ETR, _EPAUCD, _ASV, _RTR],
    [0, [() => ClientIdType, 0], 0, 1, 1, 1, () => exports.TokenValidityUnitsType$, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 0, 64 | 0, 64 | 0, 2, () => exports.AnalyticsConfigurationType$, 0, 2, 2, 1, () => exports.RefreshTokenRotationType$], 2
];
exports.UpdateUserPoolClientResponse$ = [3, n0, _UUPCRp,
    0,
    [_UPC],
    [[() => exports.UserPoolClientType$, 0]]
];
exports.UpdateUserPoolDomainRequest$ = [3, n0, _UUPDR,
    0,
    [_Do, _UPI, _MLV, _CDC],
    [0, 0, 1, () => exports.CustomDomainConfigType$], 2
];
exports.UpdateUserPoolDomainResponse$ = [3, n0, _UUPDRp,
    0,
    [_MLV, _CFD],
    [1, 0]
];
exports.UpdateUserPoolRequest$ = [3, n0, _UUPR,
    0,
    [_UPI, _Po, _DP, _LC, _AVA, _SVM, _EVM, _EVS, _VMT, _SAM, _UAUS, _MC, _DC, _EC, _SCm, _UPT, _ACUC, _UPAO, _ARS, _PNo, _UPTs],
    [0, () => exports.UserPoolPolicyType$, 0, () => exports.LambdaConfigType$, 64 | 0, 0, 0, 0, () => exports.VerificationMessageTemplateType$, 0, () => exports.UserAttributeUpdateSettingsType$, 0, () => exports.DeviceConfigurationType$, () => exports.EmailConfigurationType$, () => exports.SmsConfigurationType$, 128 | 0, () => exports.AdminCreateUserConfigType$, () => exports.UserPoolAddOnsType$, () => exports.AccountRecoverySettingType$, 0, 0], 1
];
exports.UpdateUserPoolResponse$ = [3, n0, _UUPRp,
    0,
    [],
    []
];
exports.UserAttributeUpdateSettingsType$ = [3, n0, _UAUST,
    0,
    [_ARVBU],
    [64 | 0]
];
exports.UserContextDataType$ = [3, n0, _UCDT,
    8,
    [_IA, _ED],
    [0, 0]
];
exports.UserImportJobType$ = [3, n0, _UIJT,
    0,
    [_JN, _JI, _UPI, _PSUr, _CDr, _SDt, _CDo, _St, _CWLRA, _IUm, _SUk, _FU, _CMom],
    [0, 0, 0, 0, 4, 4, 4, 0, 0, 1, 1, 1, 0]
];
exports.UsernameConfigurationType$ = [3, n0, _UCT,
    0,
    [_CSa],
    [2], 1
];
exports.UserPoolAddOnsType$ = [3, n0, _UPAOT,
    0,
    [_ASM, _ASAF],
    [0, () => exports.AdvancedSecurityAdditionalFlowsType$], 1
];
exports.UserPoolClientDescription$ = [3, n0, _UPCD,
    0,
    [_CI, _UPI, _CNl],
    [[() => ClientIdType, 0], 0, 0]
];
exports.UserPoolClientType$ = [3, n0, _UPCT,
    0,
    [_UPI, _CNl, _CI, _CSl, _LMD, _CDr, _RTV, _ATV, _ITV, _TVU, _RAe, _WA, _EAF, _SIP, _CURL, _LURL, _DRURI, _AOAF, _AOAS, _AOAFUPC, _ACn, _PUEE, _ETR, _EPAUCD, _ASV, _RTR],
    [0, 0, [() => ClientIdType, 0], [() => ClientSecretType, 0], 4, 4, 1, 1, 1, () => exports.TokenValidityUnitsType$, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 0, 64 | 0, 64 | 0, 2, () => exports.AnalyticsConfigurationType$, 0, 2, 2, 1, () => exports.RefreshTokenRotationType$]
];
exports.UserPoolDescriptionType$ = [3, n0, _UPDT,
    0,
    [_Id, _Na, _LC, _St, _LMD, _CDr],
    [0, 0, () => exports.LambdaConfigType$, 0, 4, 4]
];
exports.UserPoolPolicyType$ = [3, n0, _UPPT,
    0,
    [_PPa, _SIPi],
    [() => exports.PasswordPolicyType$, () => exports.SignInPolicyType$]
];
exports.UserPoolType$ = [3, n0, _UPTse,
    0,
    [_Id, _Na, _Po, _DP, _LC, _St, _LMD, _CDr, _SAc, _AVA, _AAl, _UAs, _SVM, _EVM, _EVS, _VMT, _SAM, _UAUS, _MC, _DC, _ENOU, _EC, _SCm, _UPT, _SCF, _ECF, _Do, _CDu, _ACUC, _UPAO, _UC, _Ar, _ARS, _UPTs],
    [0, 0, () => exports.UserPoolPolicyType$, 0, () => exports.LambdaConfigType$, 0, 4, 4, () => SchemaAttributesListType, 64 | 0, 64 | 0, 64 | 0, 0, 0, 0, () => exports.VerificationMessageTemplateType$, 0, () => exports.UserAttributeUpdateSettingsType$, 0, () => exports.DeviceConfigurationType$, 1, () => exports.EmailConfigurationType$, () => exports.SmsConfigurationType$, 128 | 0, 0, 0, 0, 0, () => exports.AdminCreateUserConfigType$, () => exports.UserPoolAddOnsType$, () => exports.UsernameConfigurationType$, 0, () => exports.AccountRecoverySettingType$, 0]
];
exports.UserType$ = [3, n0, _UTs,
    0,
    [_U, _At, _UCD, _ULMD, _E, _US, _MFAO],
    [[() => UsernameType, 0], [() => AttributeListType, 0], 4, 4, 2, 0, () => MFAOptionListType]
];
exports.VerificationMessageTemplateType$ = [3, n0, _VMTT,
    0,
    [_SM, _EM, _ESm, _EMBL, _ESBL, _DEO],
    [0, 0, 0, 0, 0, 0]
];
exports.VerifySoftwareTokenRequest$ = [3, n0, _VSTR,
    0,
    [_UCse, _ATc, _S, _FDN],
    [[() => SoftwareTokenMFAUserCodeType, 0], [() => TokenModelType, 0], [() => SessionType, 0], 0], 1
];
exports.VerifySoftwareTokenResponse$ = [3, n0, _VSTRe,
    0,
    [_St, _S],
    [0, [() => SessionType, 0]]
];
exports.VerifyUserAttributeRequest$ = [3, n0, _VUAR,
    0,
    [_ATc, _AN, _Cod],
    [[() => TokenModelType, 0], 0, 0], 3
];
exports.VerifyUserAttributeResponse$ = [3, n0, _VUARe,
    0,
    [],
    []
];
exports.WebAuthnConfigurationType$ = [3, n0, _WACT,
    0,
    [_RPI, _UV],
    [0, 0]
];
exports.WebAuthnCredentialDescription$ = [3, n0, _WACD,
    0,
    [_CIr, _FCN, _RPI, _ATu, _CAr, _AAu],
    [0, 0, 0, 64 | 0, 4, 0], 5
];
var __Unit = "unit";
var AliasAttributesListType = 64 | 0;
var AllowedFirstAuthFactorsListType = 64 | 0;
var AssetListType = [1, n0, _ALT,
    0, () => exports.AssetType$
];
var AttributeListType = [1, n0, _ALTt,
    0, [() => exports.AttributeType$,
        0]
];
var AttributeNameListType = 64 | 0;
var AttributesRequireVerificationBeforeUpdateType = 64 | 0;
var AuthEventsType = [1, n0, _AETu,
    0, () => exports.AuthEventType$
];
var AvailableChallengeListType = 64 | 0;
var BlockedIPRangeListType = 64 | 0;
var CallbackURLsListType = 64 | 0;
var ChallengeResponseListType = [1, n0, _CRLT,
    0, () => exports.ChallengeResponseType$
];
var ClientPermissionListType = 64 | 0;
var CodeDeliveryDetailsListType = [1, n0, _CDDLT,
    0, () => exports.CodeDeliveryDetailsType$
];
var ConfiguredUserAuthFactorsListType = 64 | 0;
var CustomAttributesListType = [1, n0, _CALT,
    0, () => exports.SchemaAttributeType$
];
var DeliveryMediumListType = 64 | 0;
var DeviceListType = [1, n0, _DLT,
    0, [() => exports.DeviceType$,
        0]
];
var EventFiltersType = 64 | 0;
var ExplicitAuthFlowsListType = 64 | 0;
var GroupListType = [1, n0, _GLT,
    0, () => exports.GroupType$
];
var HttpHeaderList = [1, n0, _HHL,
    0, () => exports.HttpHeader$
];
var IdpIdentifiersListType = 64 | 0;
var ListOfStringTypes = 64 | 0;
var LogConfigurationListType = [1, n0, _LCLT,
    0, () => exports.LogConfigurationType$
];
var LogoutURLsListType = 64 | 0;
var MFAOptionListType = [1, n0, _MFAOLT,
    0, () => exports.MFAOptionType$
];
var OAuthFlowsType = 64 | 0;
var ProvidersListType = [1, n0, _PLT,
    0, () => exports.ProviderDescription$
];
var RecoveryMechanismsType = [1, n0, _RMT,
    0, () => exports.RecoveryOptionType$
];
var ResourceServerScopeListType = [1, n0, _RSSLT,
    0, () => exports.ResourceServerScopeType$
];
var ResourceServersListType = [1, n0, _RSLT,
    0, () => exports.ResourceServerType$
];
var SchemaAttributesListType = [1, n0, _SALT,
    0, () => exports.SchemaAttributeType$
];
var ScopeListType = 64 | 0;
var SearchedAttributeNamesListType = 64 | 0;
var SkippedIPRangeListType = 64 | 0;
var SupportedIdentityProvidersListType = 64 | 0;
var TermsDescriptionListType = [1, n0, _TDLT,
    0, () => exports.TermsDescriptionType$
];
var UserImportJobsListType = [1, n0, _UIJLT,
    0, () => exports.UserImportJobType$
];
var UserMFASettingListType = 64 | 0;
var UsernameAttributesListType = 64 | 0;
var UserPoolClientListType = [1, n0, _UPCLT,
    0, [() => exports.UserPoolClientDescription$,
        0]
];
var UserPoolListType = [1, n0, _UPLT,
    0, () => exports.UserPoolDescriptionType$
];
var UserPoolTagsListType = 64 | 0;
var UsersListType = [1, n0, _ULT,
    0, [() => exports.UserType$,
        0]
];
var VerifiedAttributesListType = 64 | 0;
var WebAuthnAuthenticatorTransportsList = 64 | 0;
var WebAuthnCredentialDescriptionListType = [1, n0, _WACDLT,
    0, () => exports.WebAuthnCredentialDescription$
];
var AttributeMappingType = 128 | 0;
var AuthParametersType = [2, n0, _APT,
    8, 0, 0
];
var ChallengeParametersType = 128 | 0;
var ChallengeResponsesType = [2, n0, _CRTh,
    8, 0, 0
];
var ClientMetadataType = 128 | 0;
var LinksType = 128 | 0;
var ProviderDetailsType = 128 | 0;
var UserPoolTagsType = 128 | 0;
exports.AddCustomAttributes$ = [9, n0, _ACA,
    0, () => exports.AddCustomAttributesRequest$, () => exports.AddCustomAttributesResponse$
];
exports.AdminAddUserToGroup$ = [9, n0, _AAUTG,
    0, () => exports.AdminAddUserToGroupRequest$, () => __Unit
];
exports.AdminConfirmSignUp$ = [9, n0, _ACSU,
    0, () => exports.AdminConfirmSignUpRequest$, () => exports.AdminConfirmSignUpResponse$
];
exports.AdminCreateUser$ = [9, n0, _ACU,
    0, () => exports.AdminCreateUserRequest$, () => exports.AdminCreateUserResponse$
];
exports.AdminDeleteUser$ = [9, n0, _ADU,
    0, () => exports.AdminDeleteUserRequest$, () => __Unit
];
exports.AdminDeleteUserAttributes$ = [9, n0, _ADUA,
    0, () => exports.AdminDeleteUserAttributesRequest$, () => exports.AdminDeleteUserAttributesResponse$
];
exports.AdminDisableProviderForUser$ = [9, n0, _ADPFU,
    0, () => exports.AdminDisableProviderForUserRequest$, () => exports.AdminDisableProviderForUserResponse$
];
exports.AdminDisableUser$ = [9, n0, _ADUd,
    0, () => exports.AdminDisableUserRequest$, () => exports.AdminDisableUserResponse$
];
exports.AdminEnableUser$ = [9, n0, _AEU,
    0, () => exports.AdminEnableUserRequest$, () => exports.AdminEnableUserResponse$
];
exports.AdminForgetDevice$ = [9, n0, _AFD,
    0, () => exports.AdminForgetDeviceRequest$, () => __Unit
];
exports.AdminGetDevice$ = [9, n0, _AGD,
    0, () => exports.AdminGetDeviceRequest$, () => exports.AdminGetDeviceResponse$
];
exports.AdminGetUser$ = [9, n0, _AGU,
    0, () => exports.AdminGetUserRequest$, () => exports.AdminGetUserResponse$
];
exports.AdminInitiateAuth$ = [9, n0, _AIA,
    0, () => exports.AdminInitiateAuthRequest$, () => exports.AdminInitiateAuthResponse$
];
exports.AdminLinkProviderForUser$ = [9, n0, _ALPFU,
    0, () => exports.AdminLinkProviderForUserRequest$, () => exports.AdminLinkProviderForUserResponse$
];
exports.AdminListDevices$ = [9, n0, _ALD,
    0, () => exports.AdminListDevicesRequest$, () => exports.AdminListDevicesResponse$
];
exports.AdminListGroupsForUser$ = [9, n0, _ALGFU,
    0, () => exports.AdminListGroupsForUserRequest$, () => exports.AdminListGroupsForUserResponse$
];
exports.AdminListUserAuthEvents$ = [9, n0, _ALUAE,
    0, () => exports.AdminListUserAuthEventsRequest$, () => exports.AdminListUserAuthEventsResponse$
];
exports.AdminRemoveUserFromGroup$ = [9, n0, _ARUFG,
    0, () => exports.AdminRemoveUserFromGroupRequest$, () => __Unit
];
exports.AdminResetUserPassword$ = [9, n0, _ARUP,
    0, () => exports.AdminResetUserPasswordRequest$, () => exports.AdminResetUserPasswordResponse$
];
exports.AdminRespondToAuthChallenge$ = [9, n0, _ARTAC,
    0, () => exports.AdminRespondToAuthChallengeRequest$, () => exports.AdminRespondToAuthChallengeResponse$
];
exports.AdminSetUserMFAPreference$ = [9, n0, _ASUMFAP,
    0, () => exports.AdminSetUserMFAPreferenceRequest$, () => exports.AdminSetUserMFAPreferenceResponse$
];
exports.AdminSetUserPassword$ = [9, n0, _ASUP,
    0, () => exports.AdminSetUserPasswordRequest$, () => exports.AdminSetUserPasswordResponse$
];
exports.AdminSetUserSettings$ = [9, n0, _ASUS,
    0, () => exports.AdminSetUserSettingsRequest$, () => exports.AdminSetUserSettingsResponse$
];
exports.AdminUpdateAuthEventFeedback$ = [9, n0, _AUAEF,
    0, () => exports.AdminUpdateAuthEventFeedbackRequest$, () => exports.AdminUpdateAuthEventFeedbackResponse$
];
exports.AdminUpdateDeviceStatus$ = [9, n0, _AUDS,
    0, () => exports.AdminUpdateDeviceStatusRequest$, () => exports.AdminUpdateDeviceStatusResponse$
];
exports.AdminUpdateUserAttributes$ = [9, n0, _AUUA,
    0, () => exports.AdminUpdateUserAttributesRequest$, () => exports.AdminUpdateUserAttributesResponse$
];
exports.AdminUserGlobalSignOut$ = [9, n0, _AUGSO,
    0, () => exports.AdminUserGlobalSignOutRequest$, () => exports.AdminUserGlobalSignOutResponse$
];
exports.AssociateSoftwareToken$ = [9, n0, _AST,
    0, () => exports.AssociateSoftwareTokenRequest$, () => exports.AssociateSoftwareTokenResponse$
];
exports.ChangePassword$ = [9, n0, _CPh,
    0, () => exports.ChangePasswordRequest$, () => exports.ChangePasswordResponse$
];
exports.CompleteWebAuthnRegistration$ = [9, n0, _CWAR,
    0, () => exports.CompleteWebAuthnRegistrationRequest$, () => exports.CompleteWebAuthnRegistrationResponse$
];
exports.ConfirmDevice$ = [9, n0, _CDon,
    0, () => exports.ConfirmDeviceRequest$, () => exports.ConfirmDeviceResponse$
];
exports.ConfirmForgotPassword$ = [9, n0, _CFP,
    0, () => exports.ConfirmForgotPasswordRequest$, () => exports.ConfirmForgotPasswordResponse$
];
exports.ConfirmSignUp$ = [9, n0, _CSU,
    0, () => exports.ConfirmSignUpRequest$, () => exports.ConfirmSignUpResponse$
];
exports.CreateGroup$ = [9, n0, _CG,
    0, () => exports.CreateGroupRequest$, () => exports.CreateGroupResponse$
];
exports.CreateIdentityProvider$ = [9, n0, _CIP,
    0, () => exports.CreateIdentityProviderRequest$, () => exports.CreateIdentityProviderResponse$
];
exports.CreateManagedLoginBranding$ = [9, n0, _CMLB,
    0, () => exports.CreateManagedLoginBrandingRequest$, () => exports.CreateManagedLoginBrandingResponse$
];
exports.CreateResourceServer$ = [9, n0, _CRS,
    0, () => exports.CreateResourceServerRequest$, () => exports.CreateResourceServerResponse$
];
exports.CreateTerms$ = [9, n0, _CT,
    0, () => exports.CreateTermsRequest$, () => exports.CreateTermsResponse$
];
exports.CreateUserImportJob$ = [9, n0, _CUIJ,
    0, () => exports.CreateUserImportJobRequest$, () => exports.CreateUserImportJobResponse$
];
exports.CreateUserPool$ = [9, n0, _CUP,
    0, () => exports.CreateUserPoolRequest$, () => exports.CreateUserPoolResponse$
];
exports.CreateUserPoolClient$ = [9, n0, _CUPC,
    0, () => exports.CreateUserPoolClientRequest$, () => exports.CreateUserPoolClientResponse$
];
exports.CreateUserPoolDomain$ = [9, n0, _CUPD,
    0, () => exports.CreateUserPoolDomainRequest$, () => exports.CreateUserPoolDomainResponse$
];
exports.DeleteGroup$ = [9, n0, _DG,
    0, () => exports.DeleteGroupRequest$, () => __Unit
];
exports.DeleteIdentityProvider$ = [9, n0, _DIP,
    0, () => exports.DeleteIdentityProviderRequest$, () => __Unit
];
exports.DeleteManagedLoginBranding$ = [9, n0, _DMLB,
    0, () => exports.DeleteManagedLoginBrandingRequest$, () => __Unit
];
exports.DeleteResourceServer$ = [9, n0, _DRSe,
    0, () => exports.DeleteResourceServerRequest$, () => __Unit
];
exports.DeleteTerms$ = [9, n0, _DTe,
    0, () => exports.DeleteTermsRequest$, () => __Unit
];
exports.DeleteUser$ = [9, n0, _DUe,
    0, () => exports.DeleteUserRequest$, () => __Unit
];
exports.DeleteUserAttributes$ = [9, n0, _DUA,
    0, () => exports.DeleteUserAttributesRequest$, () => exports.DeleteUserAttributesResponse$
];
exports.DeleteUserPool$ = [9, n0, _DUP,
    0, () => exports.DeleteUserPoolRequest$, () => __Unit
];
exports.DeleteUserPoolClient$ = [9, n0, _DUPC,
    0, () => exports.DeleteUserPoolClientRequest$, () => __Unit
];
exports.DeleteUserPoolDomain$ = [9, n0, _DUPD,
    0, () => exports.DeleteUserPoolDomainRequest$, () => exports.DeleteUserPoolDomainResponse$
];
exports.DeleteWebAuthnCredential$ = [9, n0, _DWAC,
    0, () => exports.DeleteWebAuthnCredentialRequest$, () => exports.DeleteWebAuthnCredentialResponse$
];
exports.DescribeIdentityProvider$ = [9, n0, _DIPe,
    0, () => exports.DescribeIdentityProviderRequest$, () => exports.DescribeIdentityProviderResponse$
];
exports.DescribeManagedLoginBranding$ = [9, n0, _DMLBe,
    0, () => exports.DescribeManagedLoginBrandingRequest$, () => exports.DescribeManagedLoginBrandingResponse$
];
exports.DescribeManagedLoginBrandingByClient$ = [9, n0, _DMLBBC,
    0, () => exports.DescribeManagedLoginBrandingByClientRequest$, () => exports.DescribeManagedLoginBrandingByClientResponse$
];
exports.DescribeResourceServer$ = [9, n0, _DRSes,
    0, () => exports.DescribeResourceServerRequest$, () => exports.DescribeResourceServerResponse$
];
exports.DescribeRiskConfiguration$ = [9, n0, _DRC,
    0, () => exports.DescribeRiskConfigurationRequest$, () => exports.DescribeRiskConfigurationResponse$
];
exports.DescribeTerms$ = [9, n0, _DTes,
    0, () => exports.DescribeTermsRequest$, () => exports.DescribeTermsResponse$
];
exports.DescribeUserImportJob$ = [9, n0, _DUIJ,
    0, () => exports.DescribeUserImportJobRequest$, () => exports.DescribeUserImportJobResponse$
];
exports.DescribeUserPool$ = [9, n0, _DUPe,
    0, () => exports.DescribeUserPoolRequest$, () => exports.DescribeUserPoolResponse$
];
exports.DescribeUserPoolClient$ = [9, n0, _DUPCe,
    0, () => exports.DescribeUserPoolClientRequest$, () => exports.DescribeUserPoolClientResponse$
];
exports.DescribeUserPoolDomain$ = [9, n0, _DUPDe,
    0, () => exports.DescribeUserPoolDomainRequest$, () => exports.DescribeUserPoolDomainResponse$
];
exports.ForgetDevice$ = [9, n0, _FDo,
    0, () => exports.ForgetDeviceRequest$, () => __Unit
];
exports.ForgotPassword$ = [9, n0, _FP,
    0, () => exports.ForgotPasswordRequest$, () => exports.ForgotPasswordResponse$
];
exports.GetCSVHeader$ = [9, n0, _GCSVH,
    0, () => exports.GetCSVHeaderRequest$, () => exports.GetCSVHeaderResponse$
];
exports.GetDevice$ = [9, n0, _GD,
    0, () => exports.GetDeviceRequest$, () => exports.GetDeviceResponse$
];
exports.GetGroup$ = [9, n0, _GG,
    0, () => exports.GetGroupRequest$, () => exports.GetGroupResponse$
];
exports.GetIdentityProviderByIdentifier$ = [9, n0, _GIPBI,
    0, () => exports.GetIdentityProviderByIdentifierRequest$, () => exports.GetIdentityProviderByIdentifierResponse$
];
exports.GetLogDeliveryConfiguration$ = [9, n0, _GLDC,
    0, () => exports.GetLogDeliveryConfigurationRequest$, () => exports.GetLogDeliveryConfigurationResponse$
];
exports.GetSigningCertificate$ = [9, n0, _GSC,
    0, () => exports.GetSigningCertificateRequest$, () => exports.GetSigningCertificateResponse$
];
exports.GetTokensFromRefreshToken$ = [9, n0, _GTFRT,
    0, () => exports.GetTokensFromRefreshTokenRequest$, () => exports.GetTokensFromRefreshTokenResponse$
];
exports.GetUICustomization$ = [9, n0, _GUIC,
    0, () => exports.GetUICustomizationRequest$, () => exports.GetUICustomizationResponse$
];
exports.GetUser$ = [9, n0, _GU,
    0, () => exports.GetUserRequest$, () => exports.GetUserResponse$
];
exports.GetUserAttributeVerificationCode$ = [9, n0, _GUAVC,
    0, () => exports.GetUserAttributeVerificationCodeRequest$, () => exports.GetUserAttributeVerificationCodeResponse$
];
exports.GetUserAuthFactors$ = [9, n0, _GUAF,
    0, () => exports.GetUserAuthFactorsRequest$, () => exports.GetUserAuthFactorsResponse$
];
exports.GetUserPoolMfaConfig$ = [9, n0, _GUPMC,
    0, () => exports.GetUserPoolMfaConfigRequest$, () => exports.GetUserPoolMfaConfigResponse$
];
exports.GlobalSignOut$ = [9, n0, _GSO,
    0, () => exports.GlobalSignOutRequest$, () => exports.GlobalSignOutResponse$
];
exports.InitiateAuth$ = [9, n0, _IAn,
    0, () => exports.InitiateAuthRequest$, () => exports.InitiateAuthResponse$
];
exports.ListDevices$ = [9, n0, _LD,
    0, () => exports.ListDevicesRequest$, () => exports.ListDevicesResponse$
];
exports.ListGroups$ = [9, n0, _LG,
    0, () => exports.ListGroupsRequest$, () => exports.ListGroupsResponse$
];
exports.ListIdentityProviders$ = [9, n0, _LIP,
    0, () => exports.ListIdentityProvidersRequest$, () => exports.ListIdentityProvidersResponse$
];
exports.ListResourceServers$ = [9, n0, _LRS,
    0, () => exports.ListResourceServersRequest$, () => exports.ListResourceServersResponse$
];
exports.ListTagsForResource$ = [9, n0, _LTFR,
    0, () => exports.ListTagsForResourceRequest$, () => exports.ListTagsForResourceResponse$
];
exports.ListTerms$ = [9, n0, _LT,
    0, () => exports.ListTermsRequest$, () => exports.ListTermsResponse$
];
exports.ListUserImportJobs$ = [9, n0, _LUIJ,
    0, () => exports.ListUserImportJobsRequest$, () => exports.ListUserImportJobsResponse$
];
exports.ListUserPoolClients$ = [9, n0, _LUPC,
    0, () => exports.ListUserPoolClientsRequest$, () => exports.ListUserPoolClientsResponse$
];
exports.ListUserPools$ = [9, n0, _LUP,
    0, () => exports.ListUserPoolsRequest$, () => exports.ListUserPoolsResponse$
];
exports.ListUsers$ = [9, n0, _LU,
    0, () => exports.ListUsersRequest$, () => exports.ListUsersResponse$
];
exports.ListUsersInGroup$ = [9, n0, _LUIG,
    0, () => exports.ListUsersInGroupRequest$, () => exports.ListUsersInGroupResponse$
];
exports.ListWebAuthnCredentials$ = [9, n0, _LWAC,
    0, () => exports.ListWebAuthnCredentialsRequest$, () => exports.ListWebAuthnCredentialsResponse$
];
exports.ResendConfirmationCode$ = [9, n0, _RCC,
    0, () => exports.ResendConfirmationCodeRequest$, () => exports.ResendConfirmationCodeResponse$
];
exports.RespondToAuthChallenge$ = [9, n0, _RTAC,
    0, () => exports.RespondToAuthChallengeRequest$, () => exports.RespondToAuthChallengeResponse$
];
exports.RevokeToken$ = [9, n0, _RTev,
    0, () => exports.RevokeTokenRequest$, () => exports.RevokeTokenResponse$
];
exports.SetLogDeliveryConfiguration$ = [9, n0, _SLDC,
    0, () => exports.SetLogDeliveryConfigurationRequest$, () => exports.SetLogDeliveryConfigurationResponse$
];
exports.SetRiskConfiguration$ = [9, n0, _SRC,
    0, () => exports.SetRiskConfigurationRequest$, () => exports.SetRiskConfigurationResponse$
];
exports.SetUICustomization$ = [9, n0, _SUIC,
    0, () => exports.SetUICustomizationRequest$, () => exports.SetUICustomizationResponse$
];
exports.SetUserMFAPreference$ = [9, n0, _SUMFAP,
    0, () => exports.SetUserMFAPreferenceRequest$, () => exports.SetUserMFAPreferenceResponse$
];
exports.SetUserPoolMfaConfig$ = [9, n0, _SUPMC,
    0, () => exports.SetUserPoolMfaConfigRequest$, () => exports.SetUserPoolMfaConfigResponse$
];
exports.SetUserSettings$ = [9, n0, _SUS,
    0, () => exports.SetUserSettingsRequest$, () => exports.SetUserSettingsResponse$
];
exports.SignUp$ = [9, n0, _SUi,
    0, () => exports.SignUpRequest$, () => exports.SignUpResponse$
];
exports.StartUserImportJob$ = [9, n0, _SUIJ,
    0, () => exports.StartUserImportJobRequest$, () => exports.StartUserImportJobResponse$
];
exports.StartWebAuthnRegistration$ = [9, n0, _SWAR,
    0, () => exports.StartWebAuthnRegistrationRequest$, () => exports.StartWebAuthnRegistrationResponse$
];
exports.StopUserImportJob$ = [9, n0, _SUIJt,
    0, () => exports.StopUserImportJobRequest$, () => exports.StopUserImportJobResponse$
];
exports.TagResource$ = [9, n0, _TR,
    0, () => exports.TagResourceRequest$, () => exports.TagResourceResponse$
];
exports.UntagResource$ = [9, n0, _UR,
    0, () => exports.UntagResourceRequest$, () => exports.UntagResourceResponse$
];
exports.UpdateAuthEventFeedback$ = [9, n0, _UAEF,
    0, () => exports.UpdateAuthEventFeedbackRequest$, () => exports.UpdateAuthEventFeedbackResponse$
];
exports.UpdateDeviceStatus$ = [9, n0, _UDSp,
    0, () => exports.UpdateDeviceStatusRequest$, () => exports.UpdateDeviceStatusResponse$
];
exports.UpdateGroup$ = [9, n0, _UG,
    0, () => exports.UpdateGroupRequest$, () => exports.UpdateGroupResponse$
];
exports.UpdateIdentityProvider$ = [9, n0, _UIP,
    0, () => exports.UpdateIdentityProviderRequest$, () => exports.UpdateIdentityProviderResponse$
];
exports.UpdateManagedLoginBranding$ = [9, n0, _UMLB,
    0, () => exports.UpdateManagedLoginBrandingRequest$, () => exports.UpdateManagedLoginBrandingResponse$
];
exports.UpdateResourceServer$ = [9, n0, _URS,
    0, () => exports.UpdateResourceServerRequest$, () => exports.UpdateResourceServerResponse$
];
exports.UpdateTerms$ = [9, n0, _UTp,
    0, () => exports.UpdateTermsRequest$, () => exports.UpdateTermsResponse$
];
exports.UpdateUserAttributes$ = [9, n0, _UUA,
    0, () => exports.UpdateUserAttributesRequest$, () => exports.UpdateUserAttributesResponse$
];
exports.UpdateUserPool$ = [9, n0, _UUP,
    0, () => exports.UpdateUserPoolRequest$, () => exports.UpdateUserPoolResponse$
];
exports.UpdateUserPoolClient$ = [9, n0, _UUPC,
    0, () => exports.UpdateUserPoolClientRequest$, () => exports.UpdateUserPoolClientResponse$
];
exports.UpdateUserPoolDomain$ = [9, n0, _UUPD,
    0, () => exports.UpdateUserPoolDomainRequest$, () => exports.UpdateUserPoolDomainResponse$
];
exports.VerifySoftwareToken$ = [9, n0, _VST,
    0, () => exports.VerifySoftwareTokenRequest$, () => exports.VerifySoftwareTokenResponse$
];
exports.VerifyUserAttribute$ = [9, n0, _VUA,
    0, () => exports.VerifyUserAttributeRequest$, () => exports.VerifyUserAttributeResponse$
];
